From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alpha <alphakr93@outlook.com>
Date: Sat, 8 Jan 2022 23:22:21 +0900
Subject: [PATCH] Prismarine config files


diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index 8178d230b136121a0f5717908c344fcb9a87e71d..3045aed3fe83d1037e6a0bbefb625287c71d7ac5 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -1898,6 +1898,13 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
         }
         // Purpur end
 
+        // Prismarine start
+        @NotNull
+        public org.bukkit.configuration.file.YamlConfiguration getPrismarineConfig() {
+            throw new UnsupportedOperationException("Not supported yet.");
+        }
+        // Prismarine end
+
         /**
          * Sends the component to the player
          *
