From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alpha <alphakr93@outlook.com>
Date: Sun, 13 Feb 2022 14:20:00 +0900
Subject: [PATCH] Purpur PR - Fire Immune API


diff --git a/src/main/java/org/bukkit/entity/Entity.java b/src/main/java/org/bukkit/entity/Entity.java
index 3fa4f583c8d38c8cb62065f7f18986c985e4c92f..fe8d50118be54e41eda99688814ad6d4ee0124f2 100644
--- a/src/main/java/org/bukkit/entity/Entity.java
+++ b/src/main/java/org/bukkit/entity/Entity.java
@@ -867,5 +867,20 @@ public interface Entity extends Metadatable, CommandSender, Nameable, Persistent
      * @return True if in daylight
      */
     boolean isInDaylight();
+
+    // Prismarine start - Purpur PR
+    /**
+     * Sets if the entity is fire immune
+     * Set this to null to restore the entity type default
+     */
+    void setImmuneToFire(@Nullable Boolean fireImmune);
+
+    /**
+     * Checks if the entity is fire immune
+     *
+     * @return True if fire immune
+     */
+    boolean isImmuneToFire();
+    // Prismarine end
     // Purpur end
 }
diff --git a/src/main/java/org/bukkit/entity/Item.java b/src/main/java/org/bukkit/entity/Item.java
index 38ebf0f60113c5eff9d83e6a22ccdade651f419a..3c44dcf1f59378a1f702bbe1d624e23603189a4f 100644
--- a/src/main/java/org/bukkit/entity/Item.java
+++ b/src/main/java/org/bukkit/entity/Item.java
@@ -169,19 +169,25 @@ public interface Item extends Entity {
      */
     boolean isImmuneToExplosion();
 
+    // Prismarine start - Purpur PR
     /**
      * Set whether or not this item is immune to fire
      *
      * @param immuneToFire True to make immune to fire
+     * @deprecated use {@link #setImmuneToFire(Boolean)} instead
      */
+    @Deprecated
     void setImmuneToFire(boolean immuneToFire);
 
     /**
      * Check if item is immune to fire
      *
      * @return True if immune to fire
+     * @deprecated use Entity.isImmuneToFire() instead
      */
+    @Deprecated
     boolean isImmuneToFire();
+    // Prismarine end
 
     /**
      * Set whether or not this item is immune to lightning
