From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: AlphaKR93 <alphakr93@outlook.com>
Date: Sat, 2 Jul 2022 18:16:38 +0900
Subject: [PATCH] Purpur PR - fix #1025 (#1026)


diff --git a/src/main/java/org/bukkit/plugin/SimplePluginManager.java b/src/main/java/org/bukkit/plugin/SimplePluginManager.java
index 6c58466a6667d61b37ea1bc77652e78c4feb00db..e9b1c7a0df09b3224e356d1d0a81e3794e0ccee0 100644
--- a/src/main/java/org/bukkit/plugin/SimplePluginManager.java
+++ b/src/main/java/org/bukkit/plugin/SimplePluginManager.java
@@ -165,6 +165,15 @@ public final class SimplePluginManager implements PluginManager {
                 continue;
             }
 
+            // Prismarine start - Purpur PR
+            // Purpur start
+            if (plugins.containsKey(description.getName()) && description.getName().equalsIgnoreCase("spark")) {
+                server.getLogger().log(Level.INFO, "Purpur: Using user-provided spark plugin instead of our own.");
+                continue;
+            }
+            // Purpur end
+            // Prismarine end
+
             File replacedFile = plugins.put(description.getName(), file);
             if (replacedFile != null) {
                 server.getLogger().severe(String.format(
