From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: AlphaKR93 <alphakr93@outlook.com>
Date: Sun, 31 Jul 2022 23:18:36 +0900
Subject: [PATCH] Fix Paper patches


diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 5b1ba77a26ccf8c99891a2a2e574892c35cd4ddd..eca9ead4368e5e2822ed8c2c665902c2544815af 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -2730,11 +2730,13 @@ public final class CraftServer implements Server {
             return org.spigotmc.SpigotConfig.config;
         }
 
+        // Paper start // Prismarine - Mark
         @Override
         public YamlConfiguration getPaperConfig()
         {
             return CraftServer.this.console.paperConfigurations.createLegacyObject(CraftServer.this.console);
         }
+        // Paper end // Prismarine - Mark
 
         @Override
         public void restart() {
