From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: AlphaKR93 <alphakr93@outlook.com>
Date: Sun, 31 Jul 2022 23:18:36 +0900
Subject: [PATCH] Fix Paper patches


diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 3fffeff2f7e12c021ac4d1e6754583f58a03f12f..0accea9fab771c2a746cf49c9c4e65626fda9f45 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -2819,11 +2819,13 @@ public final class CraftServer implements Server {
             return org.spigotmc.SpigotConfig.config;
         }
 
+        // Paper start // Prismarine - Mark
         @Override
         public YamlConfiguration getPaperConfig()
         {
             return CraftServer.this.console.paperConfigurations.createLegacyObject(CraftServer.this.console);
         }
+        // Paper end // Prismarine - Mark
 
         // Purpur start
         @Override
