From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: AlphaKR93 <alphakr93@outlook.com>
Date: Sun, 31 Jul 2022 23:18:36 +0900
Subject: [PATCH] Fix Paper patches


diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 000007e77e73153d3d634a3ab5882916bd8b43a3..159582c7a27f2bed375e123aa4fcefe03f4b94f4 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -2807,11 +2807,13 @@ public final class CraftServer implements Server {
             return org.spigotmc.SpigotConfig.config;
         }
 
+        // Paper start // Prismarine - Mark
         @Override
         public YamlConfiguration getPaperConfig()
         {
             return CraftServer.this.console.paperConfigurations.createLegacyObject(CraftServer.this.console);
         }
+        // Paper end // Prismarine - Mark
 
         // Purpur start
         @Override
