From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: AlphaKR93 <alphakr93@outlook.com>
Date: Sun, 31 Jul 2022 23:18:36 +0900
Subject: [PATCH] Fix Paper patches


diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 0afdaebb071061bc26ae302bfe0d86149227850c..edc90b3d1d097040243ba372dc74c532b95c7bbd 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -2814,11 +2814,13 @@ public final class CraftServer implements Server {
             return org.spigotmc.SpigotConfig.config;
         }
 
+        // Paper start // Prismarine - Mark
         @Override
         public YamlConfiguration getPaperConfig()
         {
             return CraftServer.this.console.paperConfigurations.createLegacyObject(CraftServer.this.console);
         }
+        // Paper end // Prismarine - Mark
 
         // Purpur start
         @Override
