From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: AlphaKR93 <alphakr93@outlook.com>
Date: Thu, 28 Jul 2022 23:21:17 +0900
Subject: [PATCH] Implement No Chat Reports

Original by Aizistral Studios

diff --git a/src/main/java/net/minecraft/network/protocol/status/ServerStatus.java b/src/main/java/net/minecraft/network/protocol/status/ServerStatus.java
index 11183b4a39b9897e1b1067eb2a3556d2b1d90f52..0bd26e91b9971149a2f81f7361058038bafeb057 100644
--- a/src/main/java/net/minecraft/network/protocol/status/ServerStatus.java
+++ b/src/main/java/net/minecraft/network/protocol/status/ServerStatus.java
@@ -28,6 +28,7 @@ public class ServerStatus {
     private String favicon;
     private boolean previewsChat;
     private boolean enforcesSecureChat;
+    private boolean preventsChatReports; // Prismarine - Implement No Chat Reports
 
     @Nullable
     public Component getDescription() {
@@ -78,7 +79,7 @@ public class ServerStatus {
     }
 
     public boolean enforcesSecureChat() {
-        return this.enforcesSecureChat;
+        return (net.prismarineteam.prismarine.PrismarineConfig.disableChatReport && net.prismarineteam.prismarine.PrismarineConfig.convertToGameMessage) || this.enforcesSecureChat; // Prismarine - Implement No Chat Reports
     }
 
     public static class Players {
@@ -210,6 +211,12 @@ public class ServerStatus {
                 jsonObject.addProperty("favicon", serverStatus.getFavicon());
             }
 
+            // Prismarine start - Implement No Chat Reports
+            if (net.prismarineteam.prismarine.PrismarineConfig.disableChatReport && net.prismarineteam.prismarine.PrismarineConfig.addQueryData) {
+                jsonObject.addProperty("preventsChatReports", true);
+            }
+            // Prismarine end
+
             return jsonObject;
         }
     }
diff --git a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
index 47600c5fd0625f42a21821719ab869e273eaf807..c52618b56da668a53d0e881e1b3286372917a154 100644
--- a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
@@ -664,7 +664,7 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
 
     @Override
     public boolean enforceSecureProfile() {
-        return this.getProperties().enforceSecureProfile && this.getProperties().onlineMode;
+        return net.prismarineteam.prismarine.PrismarineConfig.disableChatReport ? false : (this.getProperties().enforceSecureProfile && this.getProperties().onlineMode); // Prismarine - Implement No Chat Reports
     }
 
     protected boolean convertOldUsers() {
diff --git a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
index d70c9d4a77edce10f783ab6c53b1e2d66b44c5fe..ed16b35a17508a04d704193885114465f44508b4 100644
--- a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -2238,10 +2238,28 @@ public class ServerGamePacketListenerImpl implements ServerPlayerConnection, Tic
 
     @Override
     public void send(Packet<?> packet) {
+        // Prismarine start - Implement No Chat Reports
+        if (net.prismarineteam.prismarine.PrismarineConfig.disableChatReport && net.prismarineteam.prismarine.PrismarineConfig.convertToGameMessage) {
+            if (packet instanceof net.minecraft.network.protocol.game.ClientboundPlayerChatHeaderPacket) return;
+            else if (packet instanceof net.minecraft.network.protocol.game.ClientboundPlayerChatPacket chat) {
+                this.send(new net.minecraft.network.protocol.game.ClientboundSystemChatPacket(chat.chatType().resolve(this.player.level.registryAccess()).get().decorate(chat.message().serverContent()), false));
+                return;
+            }
+        }
+        // Prismarine end
         this.send(packet, (PacketSendListener) null);
     }
 
     public void send(Packet<?> packet, @Nullable PacketSendListener callbacks) {
+        // Prismarine start - Implement No Chat Reports
+        if (net.prismarineteam.prismarine.PrismarineConfig.disableChatReport && net.prismarineteam.prismarine.PrismarineConfig.convertToGameMessage) {
+            if (packet instanceof net.minecraft.network.protocol.game.ClientboundPlayerChatHeaderPacket) return;
+            else if (packet instanceof net.minecraft.network.protocol.game.ClientboundPlayerChatPacket chat) {
+                this.send(chat);
+                return;
+            }
+        }
+        // Prismarine end
         // CraftBukkit start
         if (packet == null || this.processedDisconnect) { // Spigot
             return;
@@ -2640,7 +2658,7 @@ public class ServerGamePacketListenerImpl implements ServerPlayerConnection, Tic
 
     private PlayerChatMessage getSignedMessage(ServerboundChatPacket packet) {
         MessageSigner messagesigner = packet.getSigner(this.player);
-        SignedMessageChain.Link signedmessagechain_c = new SignedMessageChain.Link(packet.signature());
+        SignedMessageChain.Link signedmessagechain_c = new SignedMessageChain.Link(net.prismarineteam.prismarine.PrismarineConfig.disableChatReport ? MessageSignature.EMPTY : packet.signature()); // Prismarine - Implement No Chat Reports
         LastSeenMessages lastseenmessages = packet.lastSeenMessages().lastSeen();
         ChatMessageContent chatmessagecontent = this.getSignedContent(packet);
 
diff --git a/src/main/java/net/minecraft/server/players/PlayerList.java b/src/main/java/net/minecraft/server/players/PlayerList.java
index 9415e47e7e847116f88a0bbf4f05eed25f6b81a1..79b89424eacd770b52fb01397c1ee32ed3aaf7d5 100644
--- a/src/main/java/net/minecraft/server/players/PlayerList.java
+++ b/src/main/java/net/minecraft/server/players/PlayerList.java
@@ -1527,7 +1527,7 @@ public abstract class PlayerList {
     }
 
     public boolean verifyChatTrusted(PlayerChatMessage message, ChatSender profile) { // Paper - private -> public
-        return !message.hasExpiredServer(Instant.now()) && message.verify(profile);
+        return net.prismarineteam.prismarine.PrismarineConfig.disableChatReport || (!message.hasExpiredServer(Instant.now()) && message.verify(profile)); // Prismarine - Implement No Chat Reports
     }
 
     // CraftBukkit start
diff --git a/src/main/java/net/minecraft/world/entity/player/Player.java b/src/main/java/net/minecraft/world/entity/player/Player.java
index f669386d8b17f78006917869a8393173b676c448..4c0b68e436e34490e0b151058998f8a424859926 100644
--- a/src/main/java/net/minecraft/world/entity/player/Player.java
+++ b/src/main/java/net/minecraft/world/entity/player/Player.java
@@ -1555,7 +1555,7 @@ public abstract class Player extends LivingEntity {
 
     @Nullable
     public ProfilePublicKey getProfilePublicKey() {
-        return this.profilePublicKey;
+        return net.prismarineteam.prismarine.PrismarineConfig.disableChatReport ? null : this.profilePublicKey; // Prismarine - Implement No Chat Reports
     }
 
     public Inventory getInventory() {
diff --git a/src/main/java/net/prismarineteam/prismarine/PrismarineConfig.java b/src/main/java/net/prismarineteam/prismarine/PrismarineConfig.java
index 15d831471095782112e1087893d1bf12626e5080..a4b83873c90fa658abaeccc035778888fbc93977 100644
--- a/src/main/java/net/prismarineteam/prismarine/PrismarineConfig.java
+++ b/src/main/java/net/prismarineteam/prismarine/PrismarineConfig.java
@@ -50,8 +50,8 @@ public class PrismarineConfig {
         commands = new HashMap<>();
         commands.put("prismarine", new PrismarineCommand("prismarine"));
 
-        version = getInt("config-version", 3);
-        set("config-version", 3);
+        version = getInt("config-version", 4);
+        set("config-version", 4);
 
         readConfig(PrismarineConfig.class, null);
     }
@@ -219,4 +219,19 @@ public class PrismarineConfig {
     private static void storePlayerJsonListAsynchronously() {
         storePlayerJsonListAsynchronously = getBoolean("settings.store-player-json-list-asynchronously", storePlayerJsonListAsynchronously);
     }
+
+    public static boolean disableChatReport = false;
+    public static boolean convertToGameMessage = false;
+    public static boolean addQueryData = false;
+    private static void noChatReports() {
+        if (version < 4) {
+            set("settings.no-chat-reports.enabled", getBoolean("settings.disable-chat-report.enabled", disableChatReport));
+            set("settings.disable-chat-report.enabled", null);
+            set("settings.no-chat-reports.convert-to-game-message", getBoolean("settings.disable-chat-report.convert-to-game-message", convertToGameMessage));
+            set("settings.disable-chat-report.convert-to-game-message", null);
+        }
+        disableChatReport = getBoolean("settings.no-chat-reports.enabled", disableChatReport);
+        convertToGameMessage = getBoolean("settings.no-chat-reports.convert-to-game-message", convertToGameMessage);
+        addQueryData = getBoolean("settings.no-chat-reports.add-query-data", addQueryData);
+    }
 }
\ No newline at end of file
