From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: AlphaKR93 <alphakr93@outlook.com>
Date: Tue, 16 Aug 2022 14:49:26 +0900
Subject: [PATCH] Complete configurable entity attributes


diff --git a/src/main/java/net/minecraft/world/entity/animal/Bee.java b/src/main/java/net/minecraft/world/entity/animal/Bee.java
index b934d7d0317e6a94171ea484c72c8b15708842a3..3c9d9d1853838b36dd536662e14f8658d2aba8ed 100644
--- a/src/main/java/net/minecraft/world/entity/animal/Bee.java
+++ b/src/main/java/net/minecraft/world/entity/animal/Bee.java
@@ -218,6 +218,14 @@ public class Bee extends Animal implements NeutralMob, FlyingAnimal {
     @Override
     public void initAttributes() {
         this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level.purpurConfig.beeMaxHealth);
+        // Prismarine start
+        this.getAttribute(Attributes.FOLLOW_RANGE).setBaseValue(this.level.prismarineConfig.beeFollowRange);
+        this.getAttribute(Attributes.KNOCKBACK_RESISTANCE).setBaseValue(this.level.prismarineConfig.beeKnockbackResistance);
+        this.getAttribute(Attributes.MOVEMENT_SPEED).setBaseValue(this.level.prismarineConfig.beeMovementSpeed);
+        this.getAttribute(Attributes.ATTACK_DAMAGE).setBaseValue(this.level.prismarineConfig.beeAttackDamage);
+        this.getAttribute(Attributes.ARMOR).setBaseValue(this.level.prismarineConfig.beeArmor);
+        this.getAttribute(Attributes.ARMOR_TOUGHNESS).setBaseValue(this.level.prismarineConfig.beeArmorToughness);
+        // Prismarine end
     }
 
     @Override
diff --git a/src/main/java/net/minecraft/world/entity/animal/Cat.java b/src/main/java/net/minecraft/world/entity/animal/Cat.java
index ff3fe48b360784cf2fd63e7593487c6bc954fd09..981f7310eedc2f90fe90c92267ad24e397e502e9 100644
--- a/src/main/java/net/minecraft/world/entity/animal/Cat.java
+++ b/src/main/java/net/minecraft/world/entity/animal/Cat.java
@@ -122,6 +122,14 @@ public class Cat extends TamableAnimal {
     @Override
     public void initAttributes() {
         this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level.purpurConfig.catMaxHealth);
+        // Prismarine start
+        this.getAttribute(Attributes.FOLLOW_RANGE).setBaseValue(this.level.prismarineConfig.catFollowRange);
+        this.getAttribute(Attributes.KNOCKBACK_RESISTANCE).setBaseValue(this.level.prismarineConfig.catKnockbackResistance);
+        this.getAttribute(Attributes.MOVEMENT_SPEED).setBaseValue(this.level.prismarineConfig.catMovementSpeed);
+        this.getAttribute(Attributes.ATTACK_DAMAGE).setBaseValue(this.level.prismarineConfig.catAttackDamage);
+        this.getAttribute(Attributes.ARMOR).setBaseValue(this.level.prismarineConfig.catArmor);
+        this.getAttribute(Attributes.ARMOR_TOUGHNESS).setBaseValue(this.level.prismarineConfig.catArmorToughness);
+        // Prismarine end
     }
 
     @Override
diff --git a/src/main/java/net/minecraft/world/entity/animal/Chicken.java b/src/main/java/net/minecraft/world/entity/animal/Chicken.java
index 67893824250ca0b1f441f116aefa555b68bb594e..4de1dba80a74ef7b74436d7aa09eb09581a3b026 100644
--- a/src/main/java/net/minecraft/world/entity/animal/Chicken.java
+++ b/src/main/java/net/minecraft/world/entity/animal/Chicken.java
@@ -72,9 +72,14 @@ public class Chicken extends Animal {
     @Override
     public void initAttributes() {
         this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level.purpurConfig.chickenMaxHealth);
-        if (level.purpurConfig.chickenRetaliate) {
-            this.getAttribute(Attributes.ATTACK_DAMAGE).setBaseValue(2.0D);
-        }
+        // Prismarine start
+        this.getAttribute(Attributes.FOLLOW_RANGE).setBaseValue(this.level.prismarineConfig.chickenFollowRange);
+        this.getAttribute(Attributes.KNOCKBACK_RESISTANCE).setBaseValue(this.level.prismarineConfig.chickenKnockbackResistance);
+        this.getAttribute(Attributes.MOVEMENT_SPEED).setBaseValue(this.level.prismarineConfig.chickenMovementSpeed);
+        this.getAttribute(Attributes.ATTACK_DAMAGE).setBaseValue(this.level.prismarineConfig.chickenAttackDamage);
+        this.getAttribute(Attributes.ARMOR).setBaseValue(this.level.prismarineConfig.chickenArmor);
+        this.getAttribute(Attributes.ARMOR_TOUGHNESS).setBaseValue(this.level.prismarineConfig.chickenArmorToughness);
+        // Prismarine end
     }
 
     @Override
diff --git a/src/main/java/net/minecraft/world/entity/animal/Cod.java b/src/main/java/net/minecraft/world/entity/animal/Cod.java
index de70208403ef6c6c9c82ca4c1fd3b641a40bb45c..d1d98673261c36da79bcc02164e792ab0d218d88 100644
--- a/src/main/java/net/minecraft/world/entity/animal/Cod.java
+++ b/src/main/java/net/minecraft/world/entity/animal/Cod.java
@@ -32,6 +32,13 @@ public class Cod extends AbstractSchoolingFish {
     @Override
     public void initAttributes() {
         this.getAttribute(net.minecraft.world.entity.ai.attributes.Attributes.MAX_HEALTH).setBaseValue(this.level.purpurConfig.codMaxHealth);
+        // Prismarine start
+        this.getAttribute(net.minecraft.world.entity.ai.attributes.Attributes.FOLLOW_RANGE).setBaseValue(this.level.prismarineConfig.codFollowRange);
+        this.getAttribute(net.minecraft.world.entity.ai.attributes.Attributes.KNOCKBACK_RESISTANCE).setBaseValue(this.level.prismarineConfig.codKnockbackResistance);
+        this.getAttribute(net.minecraft.world.entity.ai.attributes.Attributes.MOVEMENT_SPEED).setBaseValue(this.level.prismarineConfig.codMovementSpeed);
+        this.getAttribute(net.minecraft.world.entity.ai.attributes.Attributes.ARMOR).setBaseValue(this.level.prismarineConfig.codArmor);
+        this.getAttribute(net.minecraft.world.entity.ai.attributes.Attributes.ARMOR_TOUGHNESS).setBaseValue(this.level.prismarineConfig.codArmorToughness);
+        // Prismarine end
     }
 
     @Override
diff --git a/src/main/java/net/minecraft/world/entity/animal/Cow.java b/src/main/java/net/minecraft/world/entity/animal/Cow.java
index a79b4a90274d25d427214b96649cdf8561469a83..aab3d8476330d94af871498f906e0c5add597ce0 100644
--- a/src/main/java/net/minecraft/world/entity/animal/Cow.java
+++ b/src/main/java/net/minecraft/world/entity/animal/Cow.java
@@ -63,6 +63,13 @@ public class Cow extends Animal {
     public void initAttributes() {
         this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level.purpurConfig.cowMaxHealth);
         this.getAttribute(Attributes.ATTACK_DAMAGE).setBaseValue(this.level.purpurConfig.cowNaturallyAggressiveToPlayersDamage); // Purpur
+        // Prismarine start
+        this.getAttribute(Attributes.FOLLOW_RANGE).setBaseValue(this.level.prismarineConfig.cowFollowRange);
+        this.getAttribute(Attributes.KNOCKBACK_RESISTANCE).setBaseValue(this.level.prismarineConfig.cowKnockbackResistance);
+        this.getAttribute(Attributes.MOVEMENT_SPEED).setBaseValue(this.level.prismarineConfig.cowMovementSpeed);
+        this.getAttribute(Attributes.ARMOR).setBaseValue(this.level.prismarineConfig.cowArmor);
+        this.getAttribute(Attributes.ARMOR_TOUGHNESS).setBaseValue(this.level.prismarineConfig.cowArmorToughness);
+        // Prismarine end
     }
 
     @Override
diff --git a/src/main/java/net/minecraft/world/entity/animal/Dolphin.java b/src/main/java/net/minecraft/world/entity/animal/Dolphin.java
index 284c1342695aeb652f39c236d14538647465846e..2b99f01b6e6a7912ea6112f00cebb832da7c37ca 100644
--- a/src/main/java/net/minecraft/world/entity/animal/Dolphin.java
+++ b/src/main/java/net/minecraft/world/entity/animal/Dolphin.java
@@ -162,6 +162,14 @@ public class Dolphin extends WaterAnimal {
     @Override
     public void initAttributes() {
         this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level.purpurConfig.dolphinMaxHealth);
+        // Prismarine start
+        this.getAttribute(Attributes.FOLLOW_RANGE).setBaseValue(this.level.prismarineConfig.dolphinFollowRange);
+        this.getAttribute(Attributes.KNOCKBACK_RESISTANCE).setBaseValue(this.level.prismarineConfig.dolphinKnockbackResistance);
+        this.getAttribute(Attributes.MOVEMENT_SPEED).setBaseValue(this.level.prismarineConfig.dolphinMovementSpeed);
+        this.getAttribute(Attributes.ATTACK_DAMAGE).setBaseValue(this.level.prismarineConfig.dolphinAttackDamage);
+        this.getAttribute(Attributes.ARMOR).setBaseValue(this.level.prismarineConfig.dolphinArmor);
+        this.getAttribute(Attributes.ARMOR_TOUGHNESS).setBaseValue(this.level.prismarineConfig.dolphinArmorToughness);
+        // Prismarine end
     }
 
     @Override
diff --git a/src/main/java/net/minecraft/world/entity/animal/Fox.java b/src/main/java/net/minecraft/world/entity/animal/Fox.java
index 21d755f228fbb49ded0e78322c6d29fc2622bb79..d55724825df08c9f7a5247b4bd98d2aa573bdef2 100644
--- a/src/main/java/net/minecraft/world/entity/animal/Fox.java
+++ b/src/main/java/net/minecraft/world/entity/animal/Fox.java
@@ -182,6 +182,14 @@ public class Fox extends Animal {
     @Override
     public void initAttributes() {
         this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level.purpurConfig.foxMaxHealth);
+        // Prismarine start
+        this.getAttribute(Attributes.FOLLOW_RANGE).setBaseValue(this.level.prismarineConfig.foxFollowRange);
+        this.getAttribute(Attributes.KNOCKBACK_RESISTANCE).setBaseValue(this.level.prismarineConfig.foxKnockbackResistance);
+        this.getAttribute(Attributes.MOVEMENT_SPEED).setBaseValue(this.level.prismarineConfig.foxMovementSpeed);
+        this.getAttribute(Attributes.ATTACK_DAMAGE).setBaseValue(this.level.prismarineConfig.foxAttackDamage);
+        this.getAttribute(Attributes.ARMOR).setBaseValue(this.level.prismarineConfig.foxArmor);
+        this.getAttribute(Attributes.ARMOR_TOUGHNESS).setBaseValue(this.level.prismarineConfig.foxArmorToughness);
+        // Prismarine end
     }
 
     @Override
diff --git a/src/main/java/net/minecraft/world/entity/animal/IronGolem.java b/src/main/java/net/minecraft/world/entity/animal/IronGolem.java
index 0292690b9c99f66210a03817e512c65ca65bc749..02c69dce92a66706d7ffb582a31d82ad6798a630 100644
--- a/src/main/java/net/minecraft/world/entity/animal/IronGolem.java
+++ b/src/main/java/net/minecraft/world/entity/animal/IronGolem.java
@@ -89,6 +89,14 @@ public class IronGolem extends AbstractGolem implements NeutralMob {
     @Override
     public void initAttributes() {
         this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level.purpurConfig.ironGolemMaxHealth);
+        // Prismarine start
+        this.getAttribute(Attributes.FOLLOW_RANGE).setBaseValue(this.level.prismarineConfig.ironGolemFollowRange);
+        this.getAttribute(Attributes.KNOCKBACK_RESISTANCE).setBaseValue(this.level.prismarineConfig.ironGolemKnockbackResistance);
+        this.getAttribute(Attributes.MOVEMENT_SPEED).setBaseValue(this.level.prismarineConfig.ironGolemMovementSpeed);
+        this.getAttribute(Attributes.ATTACK_DAMAGE).setBaseValue(this.level.prismarineConfig.ironGolemAttackDamage);
+        this.getAttribute(Attributes.ARMOR).setBaseValue(this.level.prismarineConfig.ironGolemArmor);
+        this.getAttribute(Attributes.ARMOR_TOUGHNESS).setBaseValue(this.level.prismarineConfig.ironGolemArmorToughness);
+        // Prismarine end
     }
 
     @Override
diff --git a/src/main/java/net/minecraft/world/entity/animal/MushroomCow.java b/src/main/java/net/minecraft/world/entity/animal/MushroomCow.java
index ecc36d8800427715c86060a1f29a292a475757a4..c85d54f1a5010754a528b2e54a57dab4881bc4b3 100644
--- a/src/main/java/net/minecraft/world/entity/animal/MushroomCow.java
+++ b/src/main/java/net/minecraft/world/entity/animal/MushroomCow.java
@@ -83,6 +83,14 @@ public class MushroomCow extends Cow implements Shearable {
     @Override
     public void initAttributes() {
         this.getAttribute(net.minecraft.world.entity.ai.attributes.Attributes.MAX_HEALTH).setBaseValue(this.level.purpurConfig.mooshroomMaxHealth);
+        // Prismarine start
+        this.getAttribute(net.minecraft.world.entity.ai.attributes.Attributes.FOLLOW_RANGE).setBaseValue(this.level.prismarineConfig.mooshroomFollowRange);
+        this.getAttribute(net.minecraft.world.entity.ai.attributes.Attributes.KNOCKBACK_RESISTANCE).setBaseValue(this.level.prismarineConfig.mooshroomKnockbackResistance);
+        this.getAttribute(net.minecraft.world.entity.ai.attributes.Attributes.MOVEMENT_SPEED).setBaseValue(this.level.prismarineConfig.mooshroomMovementSpeed);
+        this.getAttribute(net.minecraft.world.entity.ai.attributes.Attributes.ATTACK_DAMAGE).setBaseValue(this.level.prismarineConfig.mooshroomAttackDamage);
+        this.getAttribute(net.minecraft.world.entity.ai.attributes.Attributes.ARMOR).setBaseValue(this.level.prismarineConfig.mooshroomArmor);
+        this.getAttribute(net.minecraft.world.entity.ai.attributes.Attributes.ARMOR_TOUGHNESS).setBaseValue(this.level.prismarineConfig.mooshroomArmorToughness);
+        // Prismarine end
     }
 
     @Override
diff --git a/src/main/java/net/minecraft/world/entity/animal/Ocelot.java b/src/main/java/net/minecraft/world/entity/animal/Ocelot.java
index ded18d12b426837723f98a0163efbb2bbbbf6c00..59685074aa77e2eac1c1cd87ef600af6e4ec2c23 100644
--- a/src/main/java/net/minecraft/world/entity/animal/Ocelot.java
+++ b/src/main/java/net/minecraft/world/entity/animal/Ocelot.java
@@ -87,6 +87,14 @@ public class Ocelot extends Animal {
     @Override
     public void initAttributes() {
         this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level.purpurConfig.ocelotMaxHealth);
+        // Prismarine start
+        this.getAttribute(Attributes.FOLLOW_RANGE).setBaseValue(this.level.prismarineConfig.ocelotFollowRange);
+        this.getAttribute(Attributes.KNOCKBACK_RESISTANCE).setBaseValue(this.level.prismarineConfig.ocelotKnockbackResistance);
+        this.getAttribute(Attributes.MOVEMENT_SPEED).setBaseValue(this.level.prismarineConfig.ocelotMovementSpeed);
+        this.getAttribute(Attributes.ATTACK_DAMAGE).setBaseValue(this.level.prismarineConfig.ocelotAttackDamage);
+        this.getAttribute(Attributes.ARMOR).setBaseValue(this.level.prismarineConfig.ocelotArmor);
+        this.getAttribute(Attributes.ARMOR_TOUGHNESS).setBaseValue(this.level.prismarineConfig.ocelotArmorToughness);
+        // Prismarine end
     }
 
     @Override
diff --git a/src/main/java/net/minecraft/world/entity/animal/Panda.java b/src/main/java/net/minecraft/world/entity/animal/Panda.java
index 31a5fa82ccae53d8755a4110497c9e8c04ff347a..55d3938e6f2f20467354e4fcc47997d3c348a4b0 100644
--- a/src/main/java/net/minecraft/world/entity/animal/Panda.java
+++ b/src/main/java/net/minecraft/world/entity/animal/Panda.java
@@ -135,6 +135,14 @@ public class Panda extends Animal {
     @Override
     public void initAttributes() {
         this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level.purpurConfig.pandaMaxHealth);
+        // Prismarine start
+        this.getAttribute(Attributes.FOLLOW_RANGE).setBaseValue(this.level.prismarineConfig.pandaFollowRange);
+        this.getAttribute(Attributes.KNOCKBACK_RESISTANCE).setBaseValue(this.level.prismarineConfig.pandaKnockbackResistance);
+        this.getAttribute(Attributes.MOVEMENT_SPEED).setBaseValue(this.level.prismarineConfig.pandaMovementSpeed);
+        this.getAttribute(Attributes.ATTACK_DAMAGE).setBaseValue(this.level.prismarineConfig.pandaAttackDamage);
+        this.getAttribute(Attributes.ARMOR).setBaseValue(this.level.prismarineConfig.pandaArmor);
+        this.getAttribute(Attributes.ARMOR_TOUGHNESS).setBaseValue(this.level.prismarineConfig.pandaArmorToughness);
+        // Prismarine end
         setAttributes();
     }
 
@@ -656,7 +664,7 @@ public class Panda extends Animal {
         }
 
         if (this.isLazy()) {
-            this.getAttribute(Attributes.MOVEMENT_SPEED).setBaseValue(0.07000000029802322D);
+            this.getAttribute(Attributes.MOVEMENT_SPEED).setBaseValue(this.level.prismarineConfig.getMobAttribute(EntityType.PANDA, "lazy", Attributes.MOVEMENT_SPEED)); // Prismarine
         }
 
     }
diff --git a/src/main/java/net/minecraft/world/entity/animal/Parrot.java b/src/main/java/net/minecraft/world/entity/animal/Parrot.java
index 69fb76512ee64cbc410542db3c9969f57cad1bd4..b9f8b3c8e3d3f0d6100387112e85a5c6aac6adb4 100644
--- a/src/main/java/net/minecraft/world/entity/animal/Parrot.java
+++ b/src/main/java/net/minecraft/world/entity/animal/Parrot.java
@@ -189,6 +189,14 @@ public class Parrot extends ShoulderRidingEntity implements FlyingAnimal {
     @Override
     public void initAttributes() {
         this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level.purpurConfig.parrotMaxHealth);
+        // Prismarine start
+        this.getAttribute(Attributes.FOLLOW_RANGE).setBaseValue(this.level.prismarineConfig.parrotFollowRange);
+        this.getAttribute(Attributes.KNOCKBACK_RESISTANCE).setBaseValue(this.level.prismarineConfig.parrotKnockbackResistance);
+        this.getAttribute(Attributes.MOVEMENT_SPEED).setBaseValue(this.level.prismarineConfig.parrotMovementSpeed);
+        this.getAttribute(Attributes.ARMOR).setBaseValue(this.level.prismarineConfig.parrotArmor);
+        this.getAttribute(Attributes.ARMOR_TOUGHNESS).setBaseValue(this.level.prismarineConfig.parrotArmorToughness);
+        this.getAttribute(Attributes.FLYING_SPEED).setBaseValue(this.level.prismarineConfig.parrotFlyingSpeed);
+        // Prismarine end
     }
 
     @Override
diff --git a/src/main/java/net/minecraft/world/entity/animal/Pig.java b/src/main/java/net/minecraft/world/entity/animal/Pig.java
index f8fb169609be7cde19d88192ce6791b4fa24a738..0fc341574625e4a1931a88036c9299c79cb9bc98 100644
--- a/src/main/java/net/minecraft/world/entity/animal/Pig.java
+++ b/src/main/java/net/minecraft/world/entity/animal/Pig.java
@@ -83,6 +83,13 @@ public class Pig extends Animal implements ItemSteerable, Saddleable {
     @Override
     public void initAttributes() {
         this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level.purpurConfig.pigMaxHealth);
+        // Prismarine start
+        this.getAttribute(Attributes.FOLLOW_RANGE).setBaseValue(this.level.prismarineConfig.pigFollowRange);
+        this.getAttribute(Attributes.KNOCKBACK_RESISTANCE).setBaseValue(this.level.prismarineConfig.pigKnockbackResistance);
+        this.getAttribute(Attributes.MOVEMENT_SPEED).setBaseValue(this.level.prismarineConfig.pigMovementSpeed);
+        this.getAttribute(Attributes.ARMOR).setBaseValue(this.level.prismarineConfig.pigArmor);
+        this.getAttribute(Attributes.ARMOR_TOUGHNESS).setBaseValue(this.level.prismarineConfig.pigArmorToughness);
+        // Prismarine end
     }
 
     @Override
diff --git a/src/main/java/net/minecraft/world/entity/animal/PolarBear.java b/src/main/java/net/minecraft/world/entity/animal/PolarBear.java
index 43cc70fa6c5ad551a10054e68f6589e6514d60fc..73e4f199b66245534f1567af90d4d1df661f401e 100644
--- a/src/main/java/net/minecraft/world/entity/animal/PolarBear.java
+++ b/src/main/java/net/minecraft/world/entity/animal/PolarBear.java
@@ -96,6 +96,14 @@ public class PolarBear extends Animal implements NeutralMob {
     @Override
     public void initAttributes() {
         this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level.purpurConfig.polarBearMaxHealth);
+        // Prismarine start
+        this.getAttribute(Attributes.FOLLOW_RANGE).setBaseValue(this.level.prismarineConfig.polarBearFollowRange);
+        this.getAttribute(Attributes.KNOCKBACK_RESISTANCE).setBaseValue(this.level.prismarineConfig.polarBearKnockbackResistance);
+        this.getAttribute(Attributes.MOVEMENT_SPEED).setBaseValue(this.level.prismarineConfig.polarBearMovementSpeed);
+        this.getAttribute(Attributes.ATTACK_DAMAGE).setBaseValue(this.level.prismarineConfig.polarBearAttackDamage);
+        this.getAttribute(Attributes.ARMOR).setBaseValue(this.level.prismarineConfig.polarBearArmor);
+        this.getAttribute(Attributes.ARMOR_TOUGHNESS).setBaseValue(this.level.prismarineConfig.polarBearArmorToughness);
+        // Prismarine end
     }
 
     public boolean canMate(Animal other) {
diff --git a/src/main/java/net/minecraft/world/entity/animal/Pufferfish.java b/src/main/java/net/minecraft/world/entity/animal/Pufferfish.java
index bf9e6b6ca2b2bf8b2a2e96d10cd4fda9c59df1b8..f630f8206fa26c43eeeeba95576de3429707ab2a 100644
--- a/src/main/java/net/minecraft/world/entity/animal/Pufferfish.java
+++ b/src/main/java/net/minecraft/world/entity/animal/Pufferfish.java
@@ -64,6 +64,13 @@ public class Pufferfish extends AbstractFish {
     @Override
     public void initAttributes() {
         this.getAttribute(net.minecraft.world.entity.ai.attributes.Attributes.MAX_HEALTH).setBaseValue(this.level.purpurConfig.pufferfishMaxHealth);
+        // Prismarine start
+        this.getAttribute(net.minecraft.world.entity.ai.attributes.Attributes.FOLLOW_RANGE).setBaseValue(this.level.prismarineConfig.pufferfishFollowRange);
+        this.getAttribute(net.minecraft.world.entity.ai.attributes.Attributes.KNOCKBACK_RESISTANCE).setBaseValue(this.level.prismarineConfig.pufferfishKnockbackResistance);
+        this.getAttribute(net.minecraft.world.entity.ai.attributes.Attributes.MOVEMENT_SPEED).setBaseValue(this.level.prismarineConfig.pufferfishMovementSpeed);
+        this.getAttribute(net.minecraft.world.entity.ai.attributes.Attributes.ARMOR).setBaseValue(this.level.prismarineConfig.pufferfishArmor);
+        this.getAttribute(net.minecraft.world.entity.ai.attributes.Attributes.ARMOR_TOUGHNESS).setBaseValue(this.level.prismarineConfig.pufferfishArmorToughness);
+        // Prismarine end
     }
 
     @Override
diff --git a/src/main/java/net/minecraft/world/entity/animal/Rabbit.java b/src/main/java/net/minecraft/world/entity/animal/Rabbit.java
index e6b36650d1c670dbb3ccb0657f43432e64166e29..3556435b0f8edd559d99b2b3b6590e58de11f775 100644
--- a/src/main/java/net/minecraft/world/entity/animal/Rabbit.java
+++ b/src/main/java/net/minecraft/world/entity/animal/Rabbit.java
@@ -141,6 +141,13 @@ public class Rabbit extends Animal {
     @Override
     public void initAttributes() {
         this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level.purpurConfig.rabbitMaxHealth);
+        // Prismarine start
+        this.getAttribute(Attributes.FOLLOW_RANGE).setBaseValue(this.level.prismarineConfig.rabbitFollowRange);
+        this.getAttribute(Attributes.KNOCKBACK_RESISTANCE).setBaseValue(this.level.prismarineConfig.rabbitKnockbackResistance);
+        this.getAttribute(Attributes.MOVEMENT_SPEED).setBaseValue(this.level.prismarineConfig.rabbitMovementSpeed);
+        this.getAttribute(Attributes.ARMOR).setBaseValue(this.level.prismarineConfig.rabbitArmor);
+        this.getAttribute(Attributes.ARMOR_TOUGHNESS).setBaseValue(this.level.prismarineConfig.rabbitArmorToughness);
+        // Prismarine end
     }
 
     @Override
@@ -443,7 +450,7 @@ public class Rabbit extends Animal {
 
     public void setRabbitType(int rabbitType) {
         if (rabbitType == 99) {
-            this.getAttribute(Attributes.ARMOR).setBaseValue(8.0D);
+            this.getAttribute(Attributes.ARMOR).setBaseValue(this.level.prismarineConfig.rabbitKillerArmor); // Prismarine
             this.goalSelector.addGoal(4, new Rabbit.EvilRabbitAttackGoal(this));
             this.targetSelector.addGoal(1, (new HurtByTargetGoal(this, new Class[0])).setAlertOthers());
             this.targetSelector.addGoal(2, new NearestAttackableTargetGoal<>(this, Player.class, true));
diff --git a/src/main/java/net/minecraft/world/entity/animal/Salmon.java b/src/main/java/net/minecraft/world/entity/animal/Salmon.java
index bd1e964c7899a54a2c39afe0691a7573cfe35fc1..1513ba9a793878bef8393a49cd544db81ee4d230 100644
--- a/src/main/java/net/minecraft/world/entity/animal/Salmon.java
+++ b/src/main/java/net/minecraft/world/entity/animal/Salmon.java
@@ -32,6 +32,13 @@ public class Salmon extends AbstractSchoolingFish {
     @Override
     public void initAttributes() {
         this.getAttribute(net.minecraft.world.entity.ai.attributes.Attributes.MAX_HEALTH).setBaseValue(this.level.purpurConfig.salmonMaxHealth);
+        // Prismarine start
+        this.getAttribute(net.minecraft.world.entity.ai.attributes.Attributes.FOLLOW_RANGE).setBaseValue(this.level.prismarineConfig.salmonFollowRange);
+        this.getAttribute(net.minecraft.world.entity.ai.attributes.Attributes.KNOCKBACK_RESISTANCE).setBaseValue(this.level.prismarineConfig.salmonKnockbackResistance);
+        this.getAttribute(net.minecraft.world.entity.ai.attributes.Attributes.MOVEMENT_SPEED).setBaseValue(this.level.prismarineConfig.salmonMovementSpeed);
+        this.getAttribute(net.minecraft.world.entity.ai.attributes.Attributes.ARMOR).setBaseValue(this.level.prismarineConfig.salmonArmor);
+        this.getAttribute(net.minecraft.world.entity.ai.attributes.Attributes.ARMOR_TOUGHNESS).setBaseValue(this.level.prismarineConfig.salmonArmorToughness);
+        // Prismarine end
     }
 
     @Override
diff --git a/src/main/java/net/minecraft/world/entity/animal/Sheep.java b/src/main/java/net/minecraft/world/entity/animal/Sheep.java
index 954a12c23da96a07cd175f6f2eb28e8c5d2a8c3d..5205725e395837dd3dd1a440d30d0ef19a482875 100644
--- a/src/main/java/net/minecraft/world/entity/animal/Sheep.java
+++ b/src/main/java/net/minecraft/world/entity/animal/Sheep.java
@@ -135,6 +135,13 @@ public class Sheep extends Animal implements Shearable {
     @Override
     public void initAttributes() {
         this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level.purpurConfig.sheepMaxHealth);
+        // Prismarine start
+        this.getAttribute(Attributes.FOLLOW_RANGE).setBaseValue(this.level.prismarineConfig.sheepFollowRange);
+        this.getAttribute(Attributes.KNOCKBACK_RESISTANCE).setBaseValue(this.level.prismarineConfig.sheepKnockbackResistance);
+        this.getAttribute(Attributes.MOVEMENT_SPEED).setBaseValue(this.level.prismarineConfig.sheepMovementSpeed);
+        this.getAttribute(Attributes.ARMOR).setBaseValue(this.level.prismarineConfig.sheepArmor);
+        this.getAttribute(Attributes.ARMOR_TOUGHNESS).setBaseValue(this.level.prismarineConfig.sheepArmorToughness);
+        // Prismarine end
     }
 
     @Override
diff --git a/src/main/java/net/minecraft/world/entity/animal/SnowGolem.java b/src/main/java/net/minecraft/world/entity/animal/SnowGolem.java
index 00d86c57fabbb464a156dfaceadccd978f0d149c..8389e03cff955c8c8d05cc45d64b52b4dd537b8a 100644
--- a/src/main/java/net/minecraft/world/entity/animal/SnowGolem.java
+++ b/src/main/java/net/minecraft/world/entity/animal/SnowGolem.java
@@ -74,6 +74,14 @@ public class SnowGolem extends AbstractGolem implements Shearable, RangedAttackM
     @Override
     public void initAttributes() {
         this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level.purpurConfig.snowGolemMaxHealth);
+        // Prismarine start
+        this.getAttribute(Attributes.FOLLOW_RANGE).setBaseValue(this.level.prismarineConfig.snowGolemFollowRange);
+        this.getAttribute(Attributes.KNOCKBACK_RESISTANCE).setBaseValue(this.level.prismarineConfig.snowGolemKnockbackResistance);
+        this.getAttribute(Attributes.MOVEMENT_SPEED).setBaseValue(this.level.prismarineConfig.snowGolemMovementSpeed);
+        this.getAttribute(Attributes.ATTACK_DAMAGE).setBaseValue(this.level.prismarineConfig.snowGolemAttackDamage);
+        this.getAttribute(Attributes.ARMOR).setBaseValue(this.level.prismarineConfig.snowGolemArmor);
+        this.getAttribute(Attributes.ARMOR_TOUGHNESS).setBaseValue(this.level.prismarineConfig.snowGolemArmorToughness);
+        // Prismarine end
     }
 
     @Nullable
diff --git a/src/main/java/net/minecraft/world/entity/animal/Squid.java b/src/main/java/net/minecraft/world/entity/animal/Squid.java
index 8afdb5d4fecbb45bad2ed801fc0e526d15ef07c5..4dfb4367b59e8f3efa45136e3e508b2d4f37ae4e 100644
--- a/src/main/java/net/minecraft/world/entity/animal/Squid.java
+++ b/src/main/java/net/minecraft/world/entity/animal/Squid.java
@@ -79,6 +79,13 @@ public class Squid extends WaterAnimal {
     @Override
     public void initAttributes() {
         this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level.purpurConfig.squidMaxHealth);
+        // Prismarine start
+        this.getAttribute(Attributes.FOLLOW_RANGE).setBaseValue(this.level.prismarineConfig.squidFollowRange);
+        this.getAttribute(Attributes.KNOCKBACK_RESISTANCE).setBaseValue(this.level.prismarineConfig.squidKnockbackResistance);
+        this.getAttribute(Attributes.MOVEMENT_SPEED).setBaseValue(this.level.prismarineConfig.squidMovementSpeed);
+        this.getAttribute(Attributes.ARMOR).setBaseValue(this.level.prismarineConfig.squidArmor);
+        this.getAttribute(Attributes.ARMOR_TOUGHNESS).setBaseValue(this.level.prismarineConfig.squidArmorToughness);
+        // Prismarine end
     }
 
     @Override
diff --git a/src/main/java/net/minecraft/world/entity/animal/TropicalFish.java b/src/main/java/net/minecraft/world/entity/animal/TropicalFish.java
index 62b891c752d225f00685c42c8228eb0537f41091..6288182fc21797baece77d5b6941397000c1d06c 100644
--- a/src/main/java/net/minecraft/world/entity/animal/TropicalFish.java
+++ b/src/main/java/net/minecraft/world/entity/animal/TropicalFish.java
@@ -68,6 +68,13 @@ public class TropicalFish extends AbstractSchoolingFish {
     @Override
     public void initAttributes() {
         this.getAttribute(net.minecraft.world.entity.ai.attributes.Attributes.MAX_HEALTH).setBaseValue(this.level.purpurConfig.tropicalFishMaxHealth);
+        // Prismarine start
+        this.getAttribute(net.minecraft.world.entity.ai.attributes.Attributes.FOLLOW_RANGE).setBaseValue(this.level.prismarineConfig.tropicalFishFollowRange);
+        this.getAttribute(net.minecraft.world.entity.ai.attributes.Attributes.KNOCKBACK_RESISTANCE).setBaseValue(this.level.prismarineConfig.tropicalFishKnockbackResistance);
+        this.getAttribute(net.minecraft.world.entity.ai.attributes.Attributes.MOVEMENT_SPEED).setBaseValue(this.level.prismarineConfig.tropicalFishMovementSpeed);
+        this.getAttribute(net.minecraft.world.entity.ai.attributes.Attributes.ARMOR).setBaseValue(this.level.prismarineConfig.tropicalFishArmor);
+        this.getAttribute(net.minecraft.world.entity.ai.attributes.Attributes.ARMOR_TOUGHNESS).setBaseValue(this.level.prismarineConfig.tropicalFishArmorToughness);
+        // Prismarine end
     }
 
     @Override
diff --git a/src/main/java/net/minecraft/world/entity/animal/Turtle.java b/src/main/java/net/minecraft/world/entity/animal/Turtle.java
index e9c37c2b7fb1c6f2abc0b51c498ba8e0d2a5ef7e..d44994d5e141fa8a68a34005cf99594749b10aff 100644
--- a/src/main/java/net/minecraft/world/entity/animal/Turtle.java
+++ b/src/main/java/net/minecraft/world/entity/animal/Turtle.java
@@ -101,6 +101,13 @@ public class Turtle extends Animal {
     @Override
     public void initAttributes() {
         this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level.purpurConfig.turtleMaxHealth);
+        // Prismarine start
+        this.getAttribute(Attributes.FOLLOW_RANGE).setBaseValue(this.level.prismarineConfig.turtleFollowRange);
+        this.getAttribute(Attributes.KNOCKBACK_RESISTANCE).setBaseValue(this.level.prismarineConfig.turtleKnockbackResistance);
+        this.getAttribute(Attributes.MOVEMENT_SPEED).setBaseValue(this.level.prismarineConfig.turtleMovementSpeed);
+        this.getAttribute(Attributes.ARMOR).setBaseValue(this.level.prismarineConfig.turtleArmor);
+        this.getAttribute(Attributes.ARMOR_TOUGHNESS).setBaseValue(this.level.prismarineConfig.turtleArmorToughness);
+        // Prismarine end
     }
 
     @Override
diff --git a/src/main/java/net/minecraft/world/entity/animal/Wolf.java b/src/main/java/net/minecraft/world/entity/animal/Wolf.java
index 22a66e59687ca889965536464b628f45b0e88ec0..5008ae8fb6965a5e5aa419262cf239a607227f84 100644
--- a/src/main/java/net/minecraft/world/entity/animal/Wolf.java
+++ b/src/main/java/net/minecraft/world/entity/animal/Wolf.java
@@ -164,6 +164,14 @@ public class Wolf extends TamableAnimal implements NeutralMob {
     @Override
     public void initAttributes() {
         this.getAttribute(Attributes.MAX_HEALTH).setBaseValue(this.level.purpurConfig.wolfMaxHealth);
+        // Prismarine start
+        this.getAttribute(Attributes.FOLLOW_RANGE).setBaseValue(this.level.prismarineConfig.wolfFollowRange);
+        this.getAttribute(Attributes.KNOCKBACK_RESISTANCE).setBaseValue(this.level.prismarineConfig.wolfKnockbackResistance);
+        this.getAttribute(Attributes.MOVEMENT_SPEED).setBaseValue(this.level.prismarineConfig.wolfMovementSpeed);
+        this.getAttribute(Attributes.ATTACK_DAMAGE).setBaseValue(this.level.prismarineConfig.wolfAttackDamage);
+        this.getAttribute(Attributes.ARMOR).setBaseValue(this.level.prismarineConfig.wolfArmor);
+        this.getAttribute(Attributes.ARMOR_TOUGHNESS).setBaseValue(this.level.prismarineConfig.wolfArmorToughness);
+        // Prismarine end
     }
 
     @Override
diff --git a/src/main/java/net/prismarineteam/prismarine/PrismarineWorldConfig.java b/src/main/java/net/prismarineteam/prismarine/PrismarineWorldConfig.java
index 6b0609ff86c840355b7ec3ee14ab11db5dcbc444..1c25612d9ec87f0e1edfc3802008a3ae309d5dd3 100644
--- a/src/main/java/net/prismarineteam/prismarine/PrismarineWorldConfig.java
+++ b/src/main/java/net/prismarineteam/prismarine/PrismarineWorldConfig.java
@@ -3,10 +3,12 @@ package net.prismarineteam.prismarine;
 import org.apache.commons.lang.BooleanUtils;
 import org.bukkit.configuration.ConfigurationSection;
 
+import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.function.Predicate;
 
+import static net.prismarineteam.prismarine.PrismarineConfig.config;
 import static net.prismarineteam.prismarine.PrismarineConfig.log;
 
 @SuppressWarnings("unused")
@@ -14,6 +16,13 @@ public class PrismarineWorldConfig {
 
     private final String worldName;
 
+    private final double defaultFollowRange = 32.0D;
+    private final double defaultKnockbackResistance = 0.0D;
+    private final double defaultMovementSpeed = 0.699999988079071D;
+    private final double defaultAttackDamage = 2.0D;
+    private final double defaultArmor = 0.0D;
+    private final double defaultArmorToughness = 0.0D;
+
     public PrismarineWorldConfig(String worldName) {
         this.worldName = worldName;
         init();
@@ -59,6 +68,10 @@ public class PrismarineWorldConfig {
         return PrismarineConfig.config.getDouble("world-settings." + worldName + "." + path, PrismarineConfig.config.getDouble("world-settings.default." + path));
     }
 
+    private double getDoubleIfSet(String path, double def){
+        return PrismarineConfig.config.getDouble("world-settings." + worldName + "." + path, PrismarineConfig.config.getDouble("world-settings.default." + path, def));
+    }
+
     private int getInt(String path, int def) {
         PrismarineConfig.config.addDefault("world-settings.default." + path, def);
         return PrismarineConfig.config.getInt("world-settings." + worldName + "." + path, PrismarineConfig.config.getInt("world-settings.default." + path));
@@ -74,7 +87,7 @@ public class PrismarineWorldConfig {
         final Map<String, Object> value = PrismarineConfig.getMap("world-settings." + worldName + "." + path, null);
         return value.isEmpty() ? fallback : value;
     }
-    
+
     public double fishingRodLength = 1024.0D;
     private void fishingRodLength() {
         fishingRodLength = getDouble("fishing-rod-length", fishingRodLength);
@@ -101,4 +114,755 @@ public class PrismarineWorldConfig {
     private void timerTimeOut() {
         timerTimeOut = getInt("timer-time-out", timerTimeOut);
     }
+
+    public double beeFollowRange = 48.0D;
+    public double beeKnockbackResistance;
+    public double beeMovementSpeed = 0.30000001192092896D;
+    public double beeAttackDamage = 2.0D;
+    public double beeArmor;
+    public double beeArmorToughness;
+    public double beeFlyingSpeed = 0.6000000238418579D;
+    private void beeSettings() {
+        beeFollowRange = getDouble("mobs.bee.attribute.follow_range", beeFollowRange);
+        beeKnockbackResistance = getDouble("mobs.bee.attribute.knockback_resistance", defaultKnockbackResistance);
+        beeMovementSpeed = getDouble("mobs.bee.attribute.movement_speed", beeMovementSpeed);
+        beeAttackDamage = getDouble("mobs.bee.attribute.attack_damage", beeAttackDamage);
+        beeArmor = getDouble("mobs.bee.attribute.armor", defaultArmor);
+        beeArmorToughness = getDouble("mobs.bee.attribute.armor_toughness", defaultArmorToughness);
+        beeFlyingSpeed = getDouble("mobs.bee.attribute.flying_speed", beeFlyingSpeed);
+    }
+
+    public double chickenFollowRange;
+    public double chickenKnockbackResistance;
+    public double chickenMovementSpeed = 0.25D;
+    public double chickenAttackDamage;
+    public double chickenArmor;
+    public double chickenArmorToughness;
+    private void chickenSettings() {
+        chickenFollowRange = getDouble("mobs.chicken.attribute.follow_range", defaultFollowRange);
+        chickenKnockbackResistance = getDouble("mobs.chicken.attribute.knockback_resistance", defaultKnockbackResistance);
+        chickenMovementSpeed = getDouble("mobs.chicken.attribute.movement_speed", chickenMovementSpeed);
+        chickenAttackDamage = getDouble("mobs.chicken.attribute.attack_damage", defaultAttackDamage);
+        chickenArmor = getDouble("mobs.chicken.attribute.armor", defaultArmor);
+        chickenArmorToughness = getDouble("mobs.chicken.attribute.armor_toughness", defaultArmorToughness);
+    }
+
+    public double codFollowRange;
+    public double codKnockbackResistance;
+    public double codMovementSpeed;
+    public double codAttackDamage;
+    public double codArmor;
+    public double codArmorToughness;
+    private void codSettings() {
+        codFollowRange = getDouble("mobs.cod.attribute.follow_range", defaultFollowRange);
+        codKnockbackResistance = getDouble("mobs.cod.attribute.knockback_resistance", defaultKnockbackResistance);
+        codMovementSpeed = getDouble("mobs.cod.attribute.movement_speed", defaultMovementSpeed);
+        codAttackDamage = getDouble("mobs.cod.attribute.attack_damage", defaultAttackDamage);
+        codArmor = getDouble("mobs.cod.attribute.armor", defaultArmor);
+        codArmorToughness = getDouble("mobs.cod.attribute.armor_toughness", defaultArmorToughness);
+    }
+
+    public double cowFollowRange;
+    public double cowKnockbackResistance;
+    public double cowMovementSpeed = 0.20000000298023224D;
+    public double cowArmor;
+    public double cowArmorToughness;
+    private void cowSettings() {
+        cowFollowRange = getDouble("mobs.cow.attribute.follow_range", defaultFollowRange);
+        cowKnockbackResistance = getDouble("mobs.cow.attribute.knockback_resistance", defaultKnockbackResistance);
+        cowMovementSpeed = getDouble("mobs.cow.attribute.movement_speed", cowMovementSpeed);
+        cowArmor = getDouble("mobs.cow.attribute.armor", defaultArmor);
+        cowArmorToughness = getDouble("mobs.cow.attribute.armor_toughness", defaultArmorToughness);
+    }
+
+    public double dolphinFollowRange;
+    public double dolphinKnockbackResistance;
+    public double dolphinMovementSpeed = 1.2000000476837158D;
+    public double dolphinAttackDamage = 3.0D;
+    public double dolphinArmor;
+    public double dolphinArmorToughness;
+    private void dolphinSettings() {
+        dolphinFollowRange = getDouble("mobs.dolphin.attribute.follow_range", defaultFollowRange);
+        dolphinKnockbackResistance = getDouble("mobs.dolphin.attribute.knockback_resistance", defaultKnockbackResistance);
+        dolphinMovementSpeed = getDouble("mobs.dolphin.attribute.movement_speed", dolphinMovementSpeed);
+        dolphinAttackDamage = getDouble("mobs.dolphin.attribute.attack_damage", dolphinAttackDamage);
+        dolphinArmor = getDouble("mobs.dolphin.attribute.armor", defaultArmor);
+        dolphinArmorToughness = getDouble("mobs.dolphin.attribute.armor_toughness", defaultArmorToughness);
+    }
+
+    public double ironGolemFollowRange;
+    public double ironGolemKnockbackResistance = 1.0D;
+    public double ironGolemMovementSpeed = 0.25D;
+    public double ironGolemAttackDamage = 15.0D;
+    public double ironGolemArmor;
+    public double ironGolemArmorToughness;
+    private void ironGolemSettings() {
+        ironGolemFollowRange = getDouble("mobs.iron-golem.attribute.follow_range", defaultFollowRange);
+        ironGolemKnockbackResistance = getDouble("mobs.iron-golem.attribute.knockback_resistance", ironGolemKnockbackResistance);
+        ironGolemMovementSpeed = getDouble("mobs.iron-golem.attribute.movement_speed", ironGolemMovementSpeed);
+        ironGolemAttackDamage = getDouble("mobs.iron-golem.attribute.attack_damage", ironGolemAttackDamage);
+        ironGolemArmor = getDouble("mobs.iron-golem.attribute.armor", defaultArmor);
+        ironGolemArmorToughness = getDouble("mobs.iron-golem.attribute.armor_toughness", defaultArmorToughness);
+    }
+
+    public double ocelotFollowRange;
+    public double ocelotKnockbackResistance;
+    public double ocelotMovementSpeed = 0.30000001192092896D;
+    public double ocelotAttackDamage = 3.0D;
+    public double ocelotArmor;
+    public double ocelotArmorToughness;
+    private void ocelotSettings() {
+        ocelotFollowRange = getDouble("mobs.ocelot.attribute.follow_range", defaultFollowRange);
+        ocelotKnockbackResistance = getDouble("mobs.ocelot.attribute.knockback_resistance", defaultKnockbackResistance);
+        ocelotMovementSpeed = getDouble("mobs.ocelot.attribute.movement_speed", ocelotMovementSpeed);
+        ocelotAttackDamage = getDouble("mobs.ocelot.attribute.attack_damage", ocelotAttackDamage);
+        ocelotArmor = getDouble("mobs.ocelot.attribute.armor", defaultArmor);
+        ocelotArmorToughness = getDouble("mobs.ocelot.attribute.armor_toughness", defaultArmorToughness);
+    }
+
+    public double pigFollowRange;
+    public double pigKnockbackResistance;
+    public double pigMovementSpeed = 0.25D;
+    public double pigArmor;
+    public double pigArmorToughness;
+    private void pigSettings() {
+        pigFollowRange = getDouble("mobs.pig.attribute.follow_range", defaultFollowRange);
+        pigKnockbackResistance = getDouble("mobs.pig.attribute.knockback_resistance", defaultKnockbackResistance);
+        pigMovementSpeed = getDouble("mobs.pig.attribute.movement_speed", pigMovementSpeed);
+        pigArmor = getDouble("mobs.pig.attribute.armor", defaultArmor);
+        pigArmorToughness = getDouble("mobs.pig.attribute.armor_toughness", defaultArmorToughness);
+    }
+
+    public double polarBearFollowRange = 20.0D;
+    public double polarBearKnockbackResistance;
+    public double polarBearMovementSpeed = 0.25D;
+    public double polarBearAttackDamage = 6.0D;
+    public double polarBearArmor;
+    public double polarBearArmorToughness;
+    private void polarBearSettings() {
+        polarBearFollowRange = getDouble("mobs.polar-bear.attribute.follow_range", polarBearFollowRange);
+        polarBearKnockbackResistance = getDouble("mobs.polar-bear.attribute.knockback_resistance", defaultKnockbackResistance);
+        polarBearMovementSpeed = getDouble("mobs.polar-bear.attribute.movement_speed", polarBearMovementSpeed);
+        polarBearAttackDamage = getDouble("mobs.polar-bear.attribute.attack_damage", polarBearAttackDamage);
+        polarBearArmor = getDouble("mobs.polar-bear.attribute.armor", defaultArmor);
+        polarBearArmorToughness = getDouble("mobs.polar-bear.attribute.armor_toughness", defaultArmorToughness);
+    }
+
+    public double pufferfishFollowRange;
+    public double pufferfishKnockbackResistance;
+    public double pufferfishMovementSpeed;
+    public double pufferfishArmor;
+    public double pufferfishArmorToughness;
+    private void pufferfishSettings() {
+        pufferfishFollowRange = getDouble("mobs.pufferfish.attribute.follow_range", defaultFollowRange);
+        pufferfishKnockbackResistance = getDouble("mobs.pufferfish.attribute.knockback_resistance", defaultKnockbackResistance);
+        pufferfishMovementSpeed = getDouble("mobs.pufferfish.attribute.movement_speed", defaultMovementSpeed);
+        pufferfishArmor = getDouble("mobs.pufferfish.attribute.armor", defaultArmor);
+        pufferfishArmorToughness = getDouble("mobs.pufferfish.attribute.armor_toughness", defaultArmorToughness);
+    }
+
+    public double salmonFollowRange;
+    public double salmonKnockbackResistance;
+    public double salmonMovementSpeed;
+    public double salmonArmor;
+    public double salmonArmorToughness;
+    private void salmonSettings() {
+        salmonFollowRange = getDouble("mobs.salmon.attribute.follow_range", defaultFollowRange);
+        salmonKnockbackResistance = getDouble("mobs.salmon.attribute.knockback_resistance", defaultKnockbackResistance);
+        salmonMovementSpeed = getDouble("mobs.salmon.attribute.movement_speed", defaultMovementSpeed);
+        salmonArmor = getDouble("mobs.salmon.attribute.armor", defaultArmor);
+        salmonArmorToughness = getDouble("mobs.salmon.attribute.armor_toughness", defaultArmorToughness);
+    }
+
+    public double snowGolemFollowRange;
+    public double snowGolemKnockbackResistance;
+    public double snowGolemMovementSpeed = 0.20000000298023224D;
+    public double snowGolemAttackDamage;
+    public double snowGolemArmor;
+    public double snowGolemArmorToughness;
+    private void snowGolemSettings() {
+        snowGolemFollowRange = getDouble("mobs.snowgolem.attribute.follow_range", defaultFollowRange);
+        snowGolemKnockbackResistance = getDouble("mobs.snowgolem.attribute.knockback_resistance", defaultKnockbackResistance);
+        snowGolemMovementSpeed = getDouble("mobs.snowgolem.attribute.movement_speed", snowGolemMovementSpeed);
+        snowGolemAttackDamage = getDouble("mobs.snowgolem.attribute.attack_damage", defaultAttackDamage);
+        snowGolemArmor = getDouble("mobs.snowgolem.attribute.armor", defaultArmor);
+        snowGolemArmorToughness = getDouble("mobs.snowgolem.attribute.armor_toughness", defaultArmorToughness);
+    }
+
+    public double squidFollowRange;
+    public double squidKnockbackResistance;
+    public double squidMovementSpeed;
+    public double squidArmor;
+    public double squidArmorToughness;
+    private void squidSettings() {
+        squidFollowRange = getDouble("mobs.squid.attribute.follow_range", defaultFollowRange);
+        squidKnockbackResistance = getDouble("mobs.squid.attribute.knockback_resistance", defaultKnockbackResistance);
+        squidMovementSpeed = getDouble("mobs.squid.attribute.movement_speed", defaultMovementSpeed);
+        squidArmor = getDouble("mobs.squid.attribute.armor", defaultArmor);
+        squidArmorToughness = getDouble("mobs.squid.attribute.armor_toughness", defaultArmorToughness);
+    }
+
+    public double tropicalFishFollowRange;
+    public double tropicalFishKnockbackResistance;
+    public double tropicalFishMovementSpeed;
+    public double tropicalFishArmor;
+    public double tropicalFishArmorToughness;
+    private void tropicalFishSettings() {
+        tropicalFishFollowRange = getDouble("mobs.tropicalfish.attribute.follow_range", defaultFollowRange);
+        tropicalFishKnockbackResistance = getDouble("mobs.tropicalfish.attribute.knockback_resistance", defaultKnockbackResistance);
+        tropicalFishMovementSpeed = getDouble("mobs.tropicalfish.attribute.movement_speed", defaultMovementSpeed);
+        tropicalFishArmor = getDouble("mobs.tropicalfish.attribute.armor", defaultArmor);
+        tropicalFishArmorToughness = getDouble("mobs.tropicalfish.attribute.armor_toughness", defaultArmorToughness);
+    }
+
+    public double turtleFollowRange;
+    public double turtleKnockbackResistance;
+    public double turtleMovementSpeed = 0.25D;
+    public double turtleArmor;
+    public double turtleArmorToughness;
+    private void turtleSettings() {
+        turtleFollowRange = getDouble("mobs.turtle.attribute.follow_range", defaultFollowRange);
+        turtleKnockbackResistance = getDouble("mobs.turtle.attribute.knockback_resistance", defaultKnockbackResistance);
+        turtleMovementSpeed = getDouble("mobs.turtle.attribute.movement_speed", turtleMovementSpeed);
+        turtleArmor = getDouble("mobs.turtle.attribute.armor", defaultArmor);
+        turtleArmorToughness = getDouble("mobs.turtle.attribute.armor_toughness", defaultArmorToughness);
+    }
+
+    public double foxFollowRange;
+    public double foxKnockbackResistance;
+    public double foxMovementSpeed = 0.30000001192092896D;
+    public double foxAttackDamage;
+    public double foxArmor;
+    public double foxArmorToughness;
+    private void foxSettings() {
+        foxFollowRange = getDouble("mobs.fox.attribute.follow_range", defaultFollowRange);
+        foxKnockbackResistance = getDouble("mobs.fox.attribute.knockback_resistance", defaultKnockbackResistance);
+        foxMovementSpeed = getDouble("mobs.fox.attribute.movement_speed", foxMovementSpeed);
+        foxAttackDamage = getDouble("mobs.fox.attribute.attack_damage", defaultAttackDamage);
+        foxArmor = getDouble("mobs.fox.attribute.armor", defaultArmor);
+        foxArmorToughness = getDouble("mobs.fox.attribute.armor_toughness", defaultArmorToughness);
+    }
+
+    public double mooshroomFollowRange;
+    public double mooshroomKnockbackResistance;
+    public double mooshroomMovementSpeed;
+    public double mooshroomAttackDamage;
+    public double mooshroomArmor;
+    public double mooshroomArmorToughness;
+    private void mooshroomSettings() {
+        mooshroomFollowRange = getDouble("mobs.mooshroom.attribute.follow_range", defaultFollowRange);
+        mooshroomKnockbackResistance = getDouble("mobs.mooshroom.attribute.knockback_resistance", defaultKnockbackResistance);
+        mooshroomMovementSpeed = getDouble("mobs.mooshroom.attribute.movement_speed", cowMovementSpeed);
+        mooshroomAttackDamage = getDouble("mobs.mooshroom.attribute.attack_damage", defaultAttackDamage);
+        mooshroomArmor = getDouble("mobs.mooshroom.attribute.armor", defaultArmor);
+        mooshroomArmorToughness = getDouble("mobs.mooshroom.attribute.armor_toughness", defaultArmorToughness);
+    }
+
+    public double sheepFollowRange;
+    public double sheepKnockbackResistance;
+    public double sheepMovementSpeed = 0.23000000417232513D;
+    public double sheepArmor;
+    public double sheepArmorToughness;
+    private void sheepSettings() {
+        sheepFollowRange = getDouble("mobs.sheep.attribute.follow_range", defaultFollowRange);
+        sheepKnockbackResistance = getDouble("mobs.sheep.attribute.knockback_resistance", defaultKnockbackResistance);
+        sheepMovementSpeed = getDouble("mobs.sheep.attribute.movement_speed", sheepMovementSpeed);
+        sheepArmor = getDouble("mobs.sheep.attribute.armor", defaultArmor);
+        sheepArmorToughness = getDouble("mobs.sheep.attribute.armor_toughness", defaultArmorToughness);
+    }
+
+    public double wolfFollowRange;
+    public double wolfKnockbackResistance;
+    public double wolfMovementSpeed = 0.30000001192092896D;
+    public double wolfAttackDamage;
+    public double wolfArmor;
+    public double wolfArmorToughness;
+    private void wolfSettings() {
+        wolfFollowRange = getDouble("mobs.wolf.attribute.follow_range", defaultFollowRange);
+        wolfKnockbackResistance = getDouble("mobs.wolf.attribute.knockback_resistance", defaultKnockbackResistance);
+        wolfMovementSpeed = getDouble("mobs.wolf.attribute.movement_speed", wolfMovementSpeed);
+        wolfAttackDamage = getDouble("mobs.wolf.attribute.attack_damage", defaultAttackDamage);
+        wolfArmor = getDouble("mobs.wolf.attribute.armor", defaultArmor);
+        wolfArmorToughness = getDouble("mobs.wolf.attribute.armor_toughness", defaultArmorToughness);
+    }
+
+    private final double catMaxHealth = 10.0D;
+    public double catFollowRange;
+    public double catKnockbackResistance;
+    public double catMovementSpeed = 0.30000001192092896D;
+    public double catAttackDamage = 3.0D;
+    public double catArmor;
+    public double catArmorToughness;
+    public double catWhiteMaxHealth;
+    public double catWhiteFollowRange;
+    public double catWhiteKnockbackResistance;
+    public double catWhiteMovementSpeed;
+    public double catWhiteAttackDamage;
+    public double catWhiteArmor;
+    public double catWhiteArmorToughness;
+    public double catBlackMaxHealth;
+    public double catBlackFollowRange;
+    public double catBlackKnockbackResistance;
+    public double catBlackMovementSpeed;
+    public double catBlackAttackDamage;
+    public double catBlackArmor;
+    public double catBlackArmorToughness;
+    public double catRedMaxHealth;
+    public double catRedFollowRange;
+    public double catRedKnockbackResistance;
+    public double catRedMovementSpeed;
+    public double catRedAttackDamage;
+    public double catRedArmor;
+    public double catRedArmorToughness;
+    public double catSiameseMaxHealth;
+    public double catSiameseFollowRange;
+    public double catSiameseKnockbackResistance;
+    public double catSiameseMovementSpeed;
+    public double catSiameseAttackDamage;
+    public double catSiameseArmor;
+    public double catSiameseArmorToughness;
+    public double catBritishShorthairMaxHealth;
+    public double catBritishShorthairFollowRange;
+    public double catBritishShorthairKnockbackResistance;
+    public double catBritishShorthairMovementSpeed;
+    public double catBritishShorthairAttackDamage;
+    public double catBritishShorthairArmor;
+    public double catBritishShorthairArmorToughness;
+    public double catCalicoMaxHealth;
+    public double catCalicoFollowRange;
+    public double catCalicoKnockbackResistance;
+    public double catCalicoMovementSpeed;
+    public double catCalicoAttackDamage;
+    public double catCalicoArmor;
+    public double catCalicoArmorToughness;
+    public double catPersianMaxHealth;
+    public double catPersianFollowRange;
+    public double catPersianKnockbackResistance;
+    public double catPersianMovementSpeed;
+    public double catPersianAttackDamage;
+    public double catPersianArmor;
+    public double catPersianArmorToughness;
+    public double catRagdollMaxHealth;
+    public double catRagdollFollowRange;
+    public double catRagdollKnockbackResistance;
+    public double catRagdollMovementSpeed;
+    public double catRagdollAttackDamage;
+    public double catRagdollArmor;
+    public double catRagdollArmorToughness;
+    public double catTabbyMaxHealth;
+    public double catTabbyFollowRange;
+    public double catTabbyKnockbackResistance;
+    public double catTabbyMovementSpeed;
+    public double catTabbyAttackDamage;
+    public double catTabbyArmor;
+    public double catTabbyArmorToughness;
+    public double catAllBlackMaxHealth;
+    public double catAllBlackFollowRange;
+    public double catAllBlackKnockbackResistance;
+    public double catAllBlackMovementSpeed;
+    public double catAllBlackAttackDamage;
+    public double catAllBlackArmor;
+    public double catAllBlackArmorToughness;
+    public double catJellieMaxHealth;
+    public double catJellieFollowRange;
+    public double catJellieKnockbackResistance;
+    public double catJellieMovementSpeed;
+    public double catJellieAttackDamage;
+    public double catJellieArmor;
+    public double catJellieArmorToughness;
+    private void catSettings() {
+        catFollowRange = getDouble("mobs.cat.attribute.follow_range", defaultFollowRange);
+        catKnockbackResistance = getDouble("mobs.cat.attribute.knockback_resistance", defaultKnockbackResistance);
+        catMovementSpeed = getDouble("mobs.cat.attribute.movement_speed", catMovementSpeed);
+        catAttackDamage = getDouble("mobs.cat.attribute.attack_damage", catAttackDamage);
+        catArmor = getDouble("mobs.cat.attribute.armor", defaultArmor);
+        catArmorToughness = getDouble("mobs.cat.attribute.armor_toughness", defaultArmorToughness);
+        catWhiteMaxHealth = getDoubleIfSet("mobs.cat.attribute.white.max_health", catMaxHealth);
+        catWhiteFollowRange = getDoubleIfSet("mobs.cat.attribute.white.follow_range", defaultFollowRange);
+        catWhiteKnockbackResistance = getDoubleIfSet("mobs.cat.attribute.white.knockback_resistance", defaultKnockbackResistance);
+        catWhiteMovementSpeed = getDoubleIfSet("mobs.cat.attribute.white.movement_speed", catMovementSpeed);
+        catWhiteAttackDamage = getDoubleIfSet("mobs.cat.attribute.white.attack_damamge", catAttackDamage);
+        catWhiteArmor = getDoubleIfSet("mobs.cat.attribute.white.armor", defaultArmor);
+        catWhiteArmorToughness = getDoubleIfSet("mobs.cat.attribute.white.armor_toughness", defaultArmorToughness);
+        catBlackMaxHealth = getDoubleIfSet("mobs.cat.attribute.black.max_health", catMaxHealth);
+        catBlackFollowRange = getDoubleIfSet("mobs.cat.attribute.black.follow_range", defaultFollowRange);
+        catBlackKnockbackResistance = getDoubleIfSet("mobs.cat.attribute.black.knockback_resistance", defaultKnockbackResistance);
+        catBlackMovementSpeed = getDoubleIfSet("mobs.cat.attribute.black.movement_speed", catMovementSpeed);
+        catBlackAttackDamage = getDoubleIfSet("mobs.cat.attribute.black.attack_damamge", catAttackDamage);
+        catBlackArmor = getDoubleIfSet("mobs.cat.attribute.black.armor", defaultArmor);
+        catBlackArmorToughness = getDoubleIfSet("mobs.cat.attribute.black.armor_toughness", defaultArmorToughness);
+        catRedMaxHealth = getDoubleIfSet("mobs.cat.attribute.red.max_health", catMaxHealth);
+        catRedFollowRange = getDoubleIfSet("mobs.cat.attribute.red.follow_range", defaultFollowRange);
+        catRedKnockbackResistance = getDoubleIfSet("mobs.cat.attribute.red.knockback_resistance", defaultKnockbackResistance);
+        catRedMovementSpeed = getDoubleIfSet("mobs.cat.attribute.red.movement_speed", catMovementSpeed);
+        catRedAttackDamage = getDoubleIfSet("mobs.cat.attribute.red.attack_damamge", catAttackDamage);
+        catRedArmor = getDoubleIfSet("mobs.cat.attribute.red.armor", defaultArmor);
+        catRedArmorToughness = getDoubleIfSet("mobs.cat.attribute.red.armor_toughness", defaultArmorToughness);
+        catSiameseMaxHealth = getDoubleIfSet("mobs.cat.attribute.siamese.max_health", catMaxHealth);
+        catSiameseFollowRange = getDoubleIfSet("mobs.cat.attribute.siamese.follow_range", defaultFollowRange);
+        catSiameseKnockbackResistance = getDoubleIfSet("mobs.cat.attribute.siamese.knockback_resistance", defaultKnockbackResistance);
+        catSiameseMovementSpeed = getDoubleIfSet("mobs.cat.attribute.siamese.movement_speed", catMovementSpeed);
+        catSiameseAttackDamage = getDoubleIfSet("mobs.cat.attribute.siamese.attack_damamge", catAttackDamage);
+        catSiameseArmor = getDoubleIfSet("mobs.cat.attribute.siamese.armor", defaultArmor);
+        catSiameseArmorToughness = getDoubleIfSet("mobs.cat.attribute.siamese.armor_toughness", defaultArmorToughness);
+        catBritishShorthairMaxHealth = getDoubleIfSet("mobs.cat.attribute.british-shorthair.max_health", catMaxHealth);
+        catBritishShorthairFollowRange = getDoubleIfSet("mobs.cat.attribute.british-shorthair.follow_range", defaultFollowRange);
+        catBritishShorthairKnockbackResistance = getDoubleIfSet("mobs.cat.attribute.british-shorthair.knockback_resistance", defaultKnockbackResistance);
+        catBritishShorthairMovementSpeed = getDoubleIfSet("mobs.cat.attribute.british-shorthair.movement_speed", catMovementSpeed);
+        catBritishShorthairAttackDamage = getDoubleIfSet("mobs.cat.attribute.british-shorthair.attack_damamge", catAttackDamage);
+        catBritishShorthairArmor = getDoubleIfSet("mobs.cat.attribute.british-shorthair.armor", defaultArmor);
+        catBritishShorthairArmorToughness = getDoubleIfSet("mobs.cat.attribute.british-shorthair.armor_toughness", defaultArmorToughness);
+        catCalicoMaxHealth = getDoubleIfSet("mobs.cat.attribute.calico.max_health", catMaxHealth);
+        catCalicoFollowRange = getDoubleIfSet("mobs.cat.attribute.calico.follow_range", defaultFollowRange);
+        catCalicoKnockbackResistance = getDoubleIfSet("mobs.cat.attribute.calico.knockback_resistance", defaultKnockbackResistance);
+        catCalicoMovementSpeed = getDoubleIfSet("mobs.cat.attribute.calico.movement_speed", catMovementSpeed);
+        catCalicoAttackDamage = getDoubleIfSet("mobs.cat.attribute.calico.attack_damamge", catAttackDamage);
+        catCalicoArmor = getDoubleIfSet("mobs.cat.attribute.calico.armor", defaultArmor);
+        catCalicoArmorToughness = getDoubleIfSet("mobs.cat.attribute.calico.armor_toughness", defaultArmorToughness);
+        catPersianMaxHealth = getDoubleIfSet("mobs.cat.attribute.persian.max_health", catMaxHealth);
+        catPersianFollowRange = getDoubleIfSet("mobs.cat.attribute.persian.follow_range", defaultFollowRange);
+        catPersianKnockbackResistance = getDoubleIfSet("mobs.cat.attribute.persian.knockback_resistance", defaultKnockbackResistance);
+        catPersianMovementSpeed = getDoubleIfSet("mobs.cat.attribute.persian.movement_speed", catMovementSpeed);
+        catPersianAttackDamage = getDoubleIfSet("mobs.cat.attribute.persian.attack_damamge", catAttackDamage);
+        catPersianArmor = getDoubleIfSet("mobs.cat.attribute.persian.armor", defaultArmor);
+        catPersianArmorToughness = getDoubleIfSet("mobs.cat.attribute.persian.armor_toughness", defaultArmorToughness);
+        catRagdollMaxHealth = getDoubleIfSet("mobs.cat.attribute.ragdoll.max_health", catMaxHealth);
+        catRagdollFollowRange = getDoubleIfSet("mobs.cat.attribute.ragdoll.follow_range", defaultFollowRange);
+        catRagdollKnockbackResistance = getDoubleIfSet("mobs.cat.attribute.ragdoll.knockback_resistance", defaultKnockbackResistance);
+        catRagdollMovementSpeed = getDoubleIfSet("mobs.cat.attribute.ragdoll.movement_speed", catMovementSpeed);
+        catRagdollAttackDamage = getDoubleIfSet("mobs.cat.attribute.ragdoll.attack_damamge", catAttackDamage);
+        catRagdollArmor = getDoubleIfSet("mobs.cat.attribute.ragdoll.armor", defaultArmor);
+        catRagdollArmorToughness = getDoubleIfSet("mobs.cat.attribute.ragdoll.armor_toughness", defaultArmorToughness);
+        catTabbyMaxHealth = getDoubleIfSet("mobs.cat.attribute.tabby.max_health", catMaxHealth);
+        catTabbyFollowRange = getDoubleIfSet("mobs.cat.attribute.tabby.follow_range", defaultFollowRange);
+        catTabbyKnockbackResistance = getDoubleIfSet("mobs.cat.attribute.tabby.knockback_resistance", defaultKnockbackResistance);
+        catTabbyMovementSpeed = getDoubleIfSet("mobs.cat.attribute.tabby.movement_speed", catMovementSpeed);
+        catTabbyAttackDamage = getDoubleIfSet("mobs.cat.attribute.tabby.attack_damamge", catAttackDamage);
+        catTabbyArmor = getDoubleIfSet("mobs.cat.attribute.tabby.armor", defaultArmor);
+        catTabbyArmorToughness = getDoubleIfSet("mobs.cat.attribute.tabby.armor_toughness", defaultArmorToughness);
+        catAllBlackMaxHealth = getDoubleIfSet("mobs.cat.attribute.all-black.max_health", catMaxHealth);
+        catAllBlackFollowRange = getDoubleIfSet("mobs.cat.attribute.all-black.follow_range", defaultFollowRange);
+        catAllBlackKnockbackResistance = getDoubleIfSet("mobs.cat.attribute.all-black.knockback_resistance", defaultKnockbackResistance);
+        catAllBlackMovementSpeed = getDoubleIfSet("mobs.cat.attribute.all-black.movement_speed", catMovementSpeed);
+        catAllBlackAttackDamage = getDoubleIfSet("mobs.cat.attribute.all-black.attack_damamge", catAttackDamage);
+        catAllBlackArmor = getDoubleIfSet("mobs.cat.attribute.all-black.armor", defaultArmor);
+        catAllBlackArmorToughness = getDoubleIfSet("mobs.cat.attribute.all-black.armor_toughness", defaultArmorToughness);
+        catJellieMaxHealth = getDoubleIfSet("mobs.cat.attribute.jellie.max_health", catMaxHealth);
+        catJellieFollowRange = getDoubleIfSet("mobs.cat.attribute.jellie.follow_range", defaultFollowRange);
+        catJellieKnockbackResistance = getDoubleIfSet("mobs.cat.attribute.jellie.knockback_resistance", defaultKnockbackResistance );
+        catJellieMovementSpeed = getDoubleIfSet("mobs.cat.attribute.jellie.movement_speed", catMovementSpeed);
+        catJellieAttackDamage = getDoubleIfSet("mobs.cat.attribute.jellie.attack_damamge", catAttackDamage);
+        catJellieArmor = getDoubleIfSet("mobs.cat.attribute.jellie.armor", defaultArmor);
+        catJellieArmorToughness = getDoubleIfSet("mobs.cat.attribute.jellie.armor_toughness", defaultArmorToughness);
+    }
+
+    private final double pandaMaxHealth = 20.0D;
+    public double pandaFollowRange;
+    public double pandaKnockbackResistance;
+    public double pandaMovementSpeed = 0.15000000596046448D;
+    public double pandaAttackDamage = 6.0D;
+    public double pandaArmor;
+    public double pandaArmorToughness;
+    public double pandaNormalMaxHealth;
+    public double pandaNormalFollowRange;
+    public double pandaNormalKnockbackResistance;
+    public double pandaNormalMovementSpeed;
+    public double pandaLazyMovementSpeed = 0.07000000029802322D;
+    public double pandaNormalAttackDamage;
+    public double pandaNormalArmor;
+    public double pandaNormalArmorToughness;
+    public double pandaLazyMaxHealth;
+    public double pandaLazyFollowRange;
+    public double pandaLazyKnockbackResistance;
+    public double pandaLazyAttackDamage;
+    public double pandaLazyArmor;
+    public double pandaLazyArmorToughness;
+    public double pandaWorriedMaxHealth;
+    public double pandaWorriedFollowRange;
+    public double pandaWorriedKnockbackResistance;
+    public double pandaWorriedMovementSpeed;
+    public double pandaWorriedAttackDamage;
+    public double pandaWorriedArmor;
+    public double pandaWorriedArmorToughness;
+    public double pandaPlayfulMaxHealth;
+    public double pandaPlayfulFollowRange;
+    public double pandaPlayfulKnockbackResistance;
+    public double pandaPlayfulMovementSpeed;
+    public double pandaPlayfulAttackDamage;
+    public double pandaPlayfulArmor;
+    public double pandaPlayfulArmorToughness;
+    public double pandaBrownMaxHealth;
+    public double pandaBrownFollowRange;
+    public double pandaBrownKnockbackResistance;
+    public double pandaBrownMovementSpeed;
+    public double pandaBrownAttackDamage;
+    public double pandaBrownArmor;
+    public double pandaBrownArmorToughness;
+    public double pandaWeakMaxHealth = 10.0D;
+    public double pandaWeakFollowRange;
+    public double pandaWeakKnockbackResistance;
+    public double pandaWeakMovementSpeed;
+    public double pandaWeakAttackDamage;
+    public double pandaWeakArmor;
+    public double pandaWeakArmorToughness;
+    public double pandaAggressiveMaxHealth;
+    public double pandaAggressiveFollowRange;
+    public double pandaAggressiveKnockbackResistance;
+    public double pandaAggressiveMovementSpeed;
+    public double pandaAggressiveAttackDamage;
+    public double pandaAggressiveArmor;
+    public double pandaAggressiveArmorToughness;
+    private void pandaSettings() {
+        pandaFollowRange = getDouble("mobs.panda.normal.attribute.follow_range", defaultFollowRange);
+        pandaKnockbackResistance = getDouble("mobs.panda.normal.attribute.knockback_resistance", defaultKnockbackResistance);
+        pandaMovementSpeed = getDouble("mobs.panda.normal.attribute.movement_speed", pandaMovementSpeed);
+        pandaAttackDamage = getDouble("mobs.panda.normal.attribute.attack_damage", pandaAttackDamage);
+        pandaArmor = getDouble("mobs.panda.normal.attribute.armor", defaultArmor);
+        pandaArmorToughness = getDouble("mobs.panda.normal.attribute.armor_toughness", defaultArmorToughness);
+        pandaNormalMaxHealth = getDoubleIfSet("mobs.panda.attribute.normal.max_health", pandaMaxHealth);
+        pandaNormalFollowRange = getDoubleIfSet("mobs.panda.attribute.normal.follow_range", defaultFollowRange);
+        pandaNormalKnockbackResistance = getDoubleIfSet("mobs.panda.attribute.normal.knockback_resistance", defaultKnockbackResistance);
+        pandaNormalMovementSpeed = getDoubleIfSet("mobs.panda.attribute.normal.movement_speed", pandaMovementSpeed);
+        pandaNormalAttackDamage = getDoubleIfSet("mobs.panda.attribute.normal.attack_damage", pandaAttackDamage);
+        pandaNormalArmor = getDoubleIfSet("mobs.panda.attribute.normal.armor", defaultArmor);
+        pandaNormalArmorToughness = getDoubleIfSet("mobs.panda.attribute.normal.armor_toughness", defaultArmorToughness);
+        pandaLazyMaxHealth = getDoubleIfSet("mobs.panda.attribute.lazy.max_health", pandaMaxHealth);
+        pandaLazyFollowRange = getDoubleIfSet("mobs.panda.attribute.lazy.follow_range", defaultFollowRange);
+        pandaLazyKnockbackResistance = getDoubleIfSet("mobs.panda.attribute.lazy.knockback_resistance", defaultKnockbackResistance);
+        pandaLazyMovementSpeed = getDouble("mobs.panda.attribute.lazy.movement_speed", pandaLazyMovementSpeed);
+        pandaLazyAttackDamage = getDoubleIfSet("mobs.panda.attribute.lazy.attack_damage", pandaAttackDamage);
+        pandaLazyArmor = getDoubleIfSet("mobs.panda.attribute.lazy.armor", defaultArmor);
+        pandaLazyArmorToughness = getDoubleIfSet("mobs.panda.attribute.lazy.armor_toughness", defaultArmorToughness);
+        pandaWorriedMaxHealth = getDoubleIfSet("mobs.panda.attribute.worried.max_health", pandaMaxHealth);
+        pandaWorriedFollowRange = getDoubleIfSet("mobs.panda.attribute.worried.follow_range", defaultFollowRange);
+        pandaWorriedKnockbackResistance = getDoubleIfSet("mobs.panda.attribute.worried.knockback_resistance", defaultKnockbackResistance);
+        pandaWorriedMovementSpeed = getDoubleIfSet("mobs.panda.attribute.worried.movement_speed", pandaMovementSpeed);
+        pandaWorriedAttackDamage = getDoubleIfSet("mobs.panda.attribute.worried.attack_damage", pandaAttackDamage);
+        pandaWorriedArmor = getDoubleIfSet("mobs.panda.attribute.worried.armor", defaultArmor);
+        pandaWorriedArmorToughness = getDoubleIfSet("mobs.panda.attribute.worried.armor_toughness", defaultArmorToughness);
+        pandaPlayfulMaxHealth = getDoubleIfSet("mobs.panda.attribute.playful.max_health", pandaMaxHealth);
+        pandaPlayfulFollowRange = getDoubleIfSet("mobs.panda.attribute.playful.follow_range", defaultFollowRange);
+        pandaPlayfulKnockbackResistance = getDoubleIfSet("mobs.panda.attribute.playful.knockback_resistance", defaultKnockbackResistance);
+        pandaPlayfulMovementSpeed = getDoubleIfSet("mobs.panda.attribute.playful.movement_speed", pandaMovementSpeed);
+        pandaPlayfulAttackDamage = getDoubleIfSet("mobs.panda.attribute.playful.attack_damage", pandaAttackDamage);
+        pandaPlayfulArmor = getDoubleIfSet("mobs.panda.attribute.playful.armor", defaultArmor);
+        pandaPlayfulArmorToughness = getDoubleIfSet("mobs.panda.attribute.playful.armor_toughness", defaultArmorToughness);
+        pandaBrownMaxHealth = getDoubleIfSet("mobs.panda.attribute.brown.max_health", pandaMaxHealth);
+        pandaBrownFollowRange = getDoubleIfSet("mobs.panda.attribute.brown.follow_range", defaultFollowRange);
+        pandaBrownKnockbackResistance = getDoubleIfSet("mobs.panda.attribute.brown.knockback_resistance", defaultKnockbackResistance);
+        pandaBrownMovementSpeed = getDoubleIfSet("mobs.panda.attribute.brown.movement_speed", pandaMovementSpeed);
+        pandaBrownAttackDamage = getDoubleIfSet("mobs.panda.attribute.brown.attack_damage", pandaAttackDamage);
+        pandaBrownArmor = getDoubleIfSet("mobs.panda.attribute.brown.armor", defaultArmor);
+        pandaBrownArmorToughness = getDoubleIfSet("mobs.panda.attribute.brown.armor_toughness", defaultArmorToughness);
+        pandaWeakMaxHealth = getDouble("mobs.panda.attribute.weak.max_health", pandaWeakMaxHealth);
+        pandaWeakFollowRange = getDoubleIfSet("mobs.panda.attribute.weak.follow_range", defaultFollowRange);
+        pandaWeakKnockbackResistance = getDoubleIfSet("mobs.panda.attribute.weak.knockback_resistance", defaultKnockbackResistance);
+        pandaWeakMovementSpeed = getDoubleIfSet("mobs.panda.attribute.weak.movement_speed", pandaMovementSpeed);
+        pandaWeakAttackDamage = getDoubleIfSet("mobs.panda.attribute.weak.attack_damage", pandaAttackDamage);
+        pandaWeakArmor = getDoubleIfSet("mobs.panda.attribute.weak.armor", defaultArmor);
+        pandaWeakArmorToughness = getDoubleIfSet("mobs.panda.attribute.weak.armor_toughness", defaultArmorToughness);
+        pandaAggressiveMaxHealth = getDoubleIfSet("mobs.panda.attribute.aggressive.max_health", pandaMaxHealth);
+        pandaAggressiveFollowRange = getDoubleIfSet("mobs.panda.attribute.aggressive.follow_range", defaultFollowRange);
+        pandaAggressiveKnockbackResistance = getDoubleIfSet("mobs.panda.attribute.aggressive.knockback_resistance", defaultKnockbackResistance);
+        pandaAggressiveMovementSpeed = getDoubleIfSet("mobs.panda.attribute.aggressive.movement_speed", pandaMovementSpeed);
+        pandaAggressiveAttackDamage = getDoubleIfSet("mobs.panda.attribute.aggressive.attack_damage", pandaAttackDamage);
+        pandaAggressiveArmor = getDoubleIfSet("mobs.panda.attribute.aggressive.armor", defaultArmor);
+        pandaAggressiveArmorToughness = getDoubleIfSet("mobs.panda.attribute.aggressive.armor_toughness", defaultArmorToughness);
+    }
+
+    private final double parrotMaxHealth = 6.0D;
+    public double parrotFollowRange;
+    public double parrotKnockbackResistance;
+    public double parrotMovementSpeed = 0.20000000298023224D;
+    public double parrotArmor;
+    public double parrotArmorToughness;
+    public double parrotFlyingSpeed = 0.4000000059604645D;
+    public double parrotRedMaxHealth;
+    public double parrotRedFollowRange;
+    public double parrotRedKnockbackResistance;
+    public double parrotRedMovementSpeed;
+    public double parrotRedArmor;
+    public double parrotRedArmorToughness;
+    public double parrotRedFlyingSpeed;
+    public double parrotBlueMaxHealth;
+    public double parrotBlueFollowRange;
+    public double parrotBlueKnockbackResistance;
+    public double parrotBlueMovementSpeed;
+    public double parrotBlueArmor;
+    public double parrotBlueArmorToughness;
+    public double parrotBlueFlyingSpeed;
+    public double parrotGreenMaxHealth;
+    public double parrotGreenFollowRange;
+    public double parrotGreenKnockbackResistance;
+    public double parrotGreenMovementSpeed;
+    public double parrotGreenArmor;
+    public double parrotGreenArmorToughness;
+    public double parrotGreenFlyingSpeed;
+    public double parrotCyanMaxHealth;
+    public double parrotCyanFollowRange;
+    public double parrotCyanKnockbackResistance;
+    public double parrotCyanMovementSpeed;
+    public double parrotCyanArmor;
+    public double parrotCyanArmorToughness;
+    public double parrotCyanFlyingSpeed;
+    public double parrotGrayMaxHealth;
+    public double parrotGrayFollowRange;
+    public double parrotGrayKnockbackResistance;
+    public double parrotGrayMovementSpeed;
+    public double parrotGrayArmor;
+    public double parrotGrayArmorToughness;
+    public double parrotGrayFlyingSpeed;
+    private void parrotSettings() {
+        parrotFollowRange = getDouble("mobs.parrot.attribute.follow_range", defaultFollowRange);
+        parrotKnockbackResistance = getDouble("mobs.parrot.attribute.knockback_resistance", defaultKnockbackResistance);
+        parrotMovementSpeed = getDouble("mobs.parrot.attribute.movement_speed", parrotMovementSpeed);
+        parrotArmor = getDouble("mobs.parrot.attribute.armor", defaultArmor);
+        parrotArmorToughness = getDouble("mobs.parrot.attribute.armor_toughness", defaultArmorToughness);
+        parrotFlyingSpeed = getDouble("mobs.parrot.attribute.flying_speed", parrotFlyingSpeed);
+        parrotRedMaxHealth = getDoubleIfSet("mobs.parrot.attribute.red.max_health", parrotMaxHealth);
+        parrotRedFollowRange = getDoubleIfSet("mobs.parrot.attribute.red.follow_range", defaultFollowRange);
+        parrotRedKnockbackResistance = getDoubleIfSet("mobs.parrot.attribute.red.knockback_resistance", defaultKnockbackResistance);
+        parrotRedMovementSpeed = getDoubleIfSet("mobs.parrot.attribute.red.movement_speed", parrotMovementSpeed);
+        parrotRedArmor = getDoubleIfSet("mobs.parrot.attribute.red.armor", defaultArmor);
+        parrotRedArmorToughness = getDoubleIfSet("mobs.parrot.attribute.red.armor_toughness", defaultArmorToughness);
+        parrotRedFlyingSpeed = getDoubleIfSet("mobs.parrot.attribute.red.flying_speed", parrotFlyingSpeed);
+        parrotBlueMaxHealth = getDoubleIfSet("mobs.parrot.attribute.blue.max_health", parrotMaxHealth);
+        parrotBlueFollowRange = getDoubleIfSet("mobs.parrot.attribute.blue.follow_range", defaultFollowRange);
+        parrotBlueKnockbackResistance = getDoubleIfSet("mobs.parrot.attribute.blue.knockback_resistance", defaultKnockbackResistance);
+        parrotBlueMovementSpeed = getDoubleIfSet("mobs.parrot.attribute.blue.movement_speed", parrotMovementSpeed);
+        parrotBlueArmor = getDoubleIfSet("mobs.parrot.attribute.blue.armor", defaultArmor);
+        parrotBlueArmorToughness = getDoubleIfSet("mobs.parrot.attribute.blue.armor_toughness", defaultArmorToughness);
+        parrotBlueFlyingSpeed = getDoubleIfSet("mobs.parrot.attribute.blue.flying_speed", parrotFlyingSpeed);
+        parrotGreenMaxHealth = getDoubleIfSet("mobs.parrot.attribute.green.max_health", parrotMaxHealth);
+        parrotGreenFollowRange = getDoubleIfSet("mobs.parrot.attribute.green.follow_range", defaultFollowRange);
+        parrotGreenKnockbackResistance = getDoubleIfSet("mobs.parrot.attribute.green.knockback_resistance", defaultKnockbackResistance);
+        parrotGreenMovementSpeed = getDoubleIfSet("mobs.parrot.attribute.green.movement_speed", parrotMovementSpeed);
+        parrotGreenArmor = getDoubleIfSet("mobs.parrot.attribute.green.armor", defaultArmor);
+        parrotGreenArmorToughness = getDoubleIfSet("mobs.parrot.attribute.green.armor_toughness", defaultArmorToughness);
+        parrotGreenFlyingSpeed = getDoubleIfSet("mobs.parrot.attribute.green.flying_speed", parrotFlyingSpeed);
+        parrotCyanMaxHealth = getDoubleIfSet("mobs.parrot.attribute.cyan.max_health", parrotMaxHealth);
+        parrotCyanFollowRange = getDoubleIfSet("mobs.parrot.attribute.cyan.follow_range", defaultFollowRange);
+        parrotCyanKnockbackResistance = getDoubleIfSet("mobs.parrot.attribute.cyan.knockback_resistance", defaultKnockbackResistance);
+        parrotCyanMovementSpeed = getDoubleIfSet("mobs.parrot.attribute.cyan.movement_speed", parrotMovementSpeed);
+        parrotCyanArmor = getDoubleIfSet("mobs.parrot.attribute.cyan.armor", defaultArmor);
+        parrotCyanArmorToughness = getDoubleIfSet("mobs.parrot.attribute.cyan.armor_toughness", defaultArmorToughness);
+        parrotCyanFlyingSpeed = getDoubleIfSet("mobs.parrot.attribute.cyan.flying_speed", parrotFlyingSpeed);
+        parrotGrayMaxHealth = getDoubleIfSet("mobs.parrot.attribute.gray.max_health", parrotMaxHealth);
+        parrotGrayFollowRange = getDoubleIfSet("mobs.parrot.attribute.gray.follow_range", defaultFollowRange);
+        parrotGrayKnockbackResistance = getDoubleIfSet("mobs.parrot.attribute.gray.knockback_resistance", defaultKnockbackResistance);
+        parrotGrayMovementSpeed = getDoubleIfSet("mobs.parrot.attribute.gray.movement_speed", parrotMovementSpeed);
+        parrotGrayArmor = getDoubleIfSet("mobs.parrot.attribute.gray.armor", defaultArmor);
+        parrotGrayArmorToughness = getDoubleIfSet("mobs.parrot.attribute.gray.armor_toughness", defaultArmorToughness);
+        parrotGrayFlyingSpeed = getDoubleIfSet("mobs.parrot.attribute.gray.flying_speed", parrotFlyingSpeed);
+    }
+
+    private final double rabbitMaxHealth = 3.0D;
+    public double rabbitFollowRange;
+    public double rabbitKnockbackResistance;
+    public double rabbitMovementSpeed = 0.30000001192092896D;
+    public double rabbitArmor;
+    public double rabbitArmorToughness;
+    public double rabbitBrownMaxHealth;
+    public double rabbitBrownFollowRange;
+    public double rabbitBrownKnockbackResistance;
+    public double rabbitBrownMovementSpeed;
+    public double rabbitBrownArmor;
+    public double rabbitBrownArmorToughness;
+    public double rabbitWhiteMaxHealth;
+    public double rabbitWhiteFollowRange;
+    public double rabbitWhiteKnockbackResistance;
+    public double rabbitWhiteMovementSpeed;
+    public double rabbitWhiteArmor;
+    public double rabbitWhiteArmorToughness;
+    public double rabbitBlackMaxHealth;
+    public double rabbitBlackFollowRange;
+    public double rabbitBlackKnockbackResistance;
+    public double rabbitBlackMovementSpeed;
+    public double rabbitBlackArmor;
+    public double rabbitBlackArmorToughness;
+    public double rabbitBlackAndWhiteMaxHealth;
+    public double rabbitBlackAndWhiteFollowRange;
+    public double rabbitBlackAndWhiteKnockbackResistance;
+    public double rabbitBlackAndWhiteMovementSpeed;
+    public double rabbitBlackAndWhiteArmor;
+    public double rabbitBlackAndWhiteArmorToughness;
+    public double rabbitGoldMaxHealth;
+    public double rabbitGoldFollowRange;
+    public double rabbitGoldKnockbackResistance;
+    public double rabbitGoldMovementSpeed;
+    public double rabbitGoldArmor;
+    public double rabbitGoldArmorToughness;
+    public double rabbitSaltAndPepperMaxHealth;
+    public double rabbitSaltAndPepperFollowRange;
+    public double rabbitSaltAndPepperKnockbackResistance;
+    public double rabbitSaltAndPepperMovementSpeed;
+    public double rabbitSaltAndPepperArmor;
+    public double rabbitSaltAndPepperArmorToughness;
+    public double rabbitKillerMaxHealth;
+    public double rabbitKillerFollowRange;
+    public double rabbitKillerKnockbackResistance;
+    public double rabbitKillerMovementSpeed;
+    public double rabbitKillerArmor = 8.0D;
+    public double rabbitKillerArmorToughness;
+    public double rabbitToastMaxHealth;
+    public double rabbitToastFollowRange;
+    public double rabbitToastKnockbackResistance;
+    public double rabbitToastMovementSpeed;
+    public double rabbitToastArmor;
+    public double rabbitToastArmorToughness;
+    private void rabbitSettings() {
+        rabbitFollowRange = getDouble("mobs.rabbit.attribute.default.follow_range", defaultFollowRange);
+        rabbitKnockbackResistance = getDouble("mobs.rabbit.attribute.default.knockback_resistance", defaultKnockbackResistance);
+        rabbitMovementSpeed = getDouble("mobs.rabbit.attribute.default.movement_speed", rabbitMovementSpeed);
+        rabbitArmor = getDouble("mobs.rabbit.attribute.default.armor", defaultArmor);
+        rabbitArmorToughness = getDouble("mobs.rabbit.attribute.default.armor_toughness", defaultArmorToughness);
+        rabbitBrownMaxHealth = getDoubleIfSet("mobs.rabbit.attribute.brown.max_health", rabbitMaxHealth);
+        rabbitBrownFollowRange = getDoubleIfSet("mobs.rabbit.attribute.brown.follow_range", rabbitFollowRange);
+        rabbitBrownKnockbackResistance = getDoubleIfSet("mobs.rabbit.attribute.brown.knockback_resistance", rabbitKnockbackResistance);
+        rabbitBrownMovementSpeed = getDoubleIfSet("mobs.rabbit.attribute.brown.movement_speed", rabbitMovementSpeed);
+        rabbitBrownArmor = getDoubleIfSet("mobs.rabbit.attribute.brown.armor", rabbitArmor);
+        rabbitBrownArmorToughness = getDoubleIfSet("mobs.rabbit.attribute.brown.armor_toughness", rabbitArmorToughness);
+        rabbitWhiteMaxHealth = getDoubleIfSet("mobs.rabbit.attribute.white.max_health", rabbitMaxHealth);
+        rabbitWhiteFollowRange = getDoubleIfSet("mobs.rabbit.attribute.white.follow_range", rabbitFollowRange);
+        rabbitWhiteKnockbackResistance = getDoubleIfSet("mobs.rabbit.attribute.white.knockback_resistance", rabbitKnockbackResistance);
+        rabbitWhiteMovementSpeed = getDoubleIfSet("mobs.rabbit.attribute.white.movement_speed", rabbitMovementSpeed);
+        rabbitWhiteArmor = getDoubleIfSet("mobs.rabbit.attribute.white.armor", rabbitArmor);
+        rabbitWhiteArmorToughness = getDoubleIfSet("mobs.rabbit.attribute.white.armor_toughness", rabbitArmorToughness);
+        rabbitBlackMaxHealth = getDoubleIfSet("mobs.rabbit.attribute.black.max_health", rabbitMaxHealth);
+        rabbitBlackFollowRange = getDoubleIfSet("mobs.rabbit.attribute.black.follow_range", rabbitFollowRange);
+        rabbitBlackKnockbackResistance = getDoubleIfSet("mobs.rabbit.attribute.black.knockback_resistance", rabbitKnockbackResistance);
+        rabbitBlackMovementSpeed = getDoubleIfSet("mobs.rabbit.attribute.black.movement_speed", rabbitMovementSpeed);
+        rabbitBlackArmor = getDoubleIfSet("mobs.rabbit.attribute.black.armor", rabbitArmor);
+        rabbitBlackArmorToughness = getDoubleIfSet("mobs.rabbit.attribute.black.armor_toughness", rabbitArmorToughness);
+        rabbitBlackAndWhiteMaxHealth = getDoubleIfSet("mobs.rabbit.attribute.black_and_white.max_health", rabbitMaxHealth);
+        rabbitBlackAndWhiteFollowRange = getDoubleIfSet("mobs.rabbit.attribute.black_and_white.follow_range", rabbitFollowRange);
+        rabbitBlackAndWhiteKnockbackResistance = getDoubleIfSet("mobs.rabbit.attribute.black_and_white.knockback_resistance", rabbitKnockbackResistance);
+        rabbitBlackAndWhiteMovementSpeed = getDoubleIfSet("mobs.rabbit.attribute.black_and_white.movement_speed", rabbitMovementSpeed);
+        rabbitBlackAndWhiteArmor = getDoubleIfSet("mobs.rabbit.attribute.black_and_white.armor", rabbitArmor);
+        rabbitBlackAndWhiteArmorToughness = getDoubleIfSet("mobs.rabbit.attribute.black_and_white.armor_toughness", rabbitArmorToughness);
+        rabbitGoldMaxHealth = getDoubleIfSet("mobs.rabbit.attribute.gold.max_health", rabbitMaxHealth);
+        rabbitGoldFollowRange = getDoubleIfSet("mobs.rabbit.attribute.gold.follow_range", rabbitFollowRange);
+        rabbitGoldKnockbackResistance = getDoubleIfSet("mobs.rabbit.attribute.gold.knockback_resistance", rabbitKnockbackResistance);
+        rabbitGoldMovementSpeed = getDoubleIfSet("mobs.rabbit.attribute.gold.movement_speed", rabbitMovementSpeed);
+        rabbitGoldArmor = getDoubleIfSet("mobs.rabbit.attribute.gold.armor", rabbitArmor);
+        rabbitGoldArmorToughness = getDoubleIfSet("mobs.rabbit.attribute.gold.armor_toughness", rabbitArmorToughness);
+        rabbitSaltAndPepperMaxHealth = getDoubleIfSet("mobs.rabbit.attribute.slat_and_pepper.max_health", rabbitMaxHealth);
+        rabbitSaltAndPepperFollowRange = getDoubleIfSet("mobs.rabbit.attribute.slat_and_pepper.follow_range", rabbitFollowRange);
+        rabbitSaltAndPepperKnockbackResistance = getDoubleIfSet("mobs.rabbit.attribute.slat_and_pepper.knockback_resistance", rabbitKnockbackResistance);
+        rabbitSaltAndPepperMovementSpeed = getDoubleIfSet("mobs.rabbit.attribute.slat_and_pepper.movement_speed", rabbitMovementSpeed);
+        rabbitSaltAndPepperArmor = getDoubleIfSet("mobs.rabbit.attribute.slat_and_pepper.armor", rabbitArmor);
+        rabbitSaltAndPepperArmorToughness = getDoubleIfSet("mobs.rabbit.attribute.slat_and_pepper.armor_toughness", rabbitArmorToughness);
+        rabbitKillerMaxHealth = getDoubleIfSet("mobs.rabbit.attribute.killer.max_health", rabbitMaxHealth);
+        rabbitKillerFollowRange = getDoubleIfSet("mobs.rabbit.attribute.killer.follow_range", defaultFollowRange);
+        rabbitKillerKnockbackResistance = getDoubleIfSet("mobs.rabbit.attribute.killer.knockback_resistance", defaultKnockbackResistance);
+        rabbitKillerMovementSpeed = getDoubleIfSet("mobs.rabbit.attribute.killer.movement_speed", rabbitMovementSpeed);
+        rabbitKillerArmor = getDouble("mobs.rabbit.attribute.killer.armor", rabbitKillerArmor);
+        rabbitKillerArmorToughness = getDoubleIfSet("mobs.rabbit.attribute.killer.armor_toughness", defaultArmorToughness);
+        rabbitToastMaxHealth = getDoubleIfSet("mobs.rabbit.attribute.toast.max_health", rabbitMaxHealth);
+        rabbitToastFollowRange = getDoubleIfSet("mobs.rabbit.attribute.toast.follow_range", rabbitFollowRange);
+        rabbitToastKnockbackResistance = getDoubleIfSet("mobs.rabbit.attribute.toast.knockback_resistance", rabbitKnockbackResistance);
+        rabbitToastMovementSpeed = getDoubleIfSet("mobs.rabbit.attribute.toast.movement_speed", rabbitMovementSpeed);
+        rabbitToastArmor = getDoubleIfSet("mobs.rabbit.attribute.toast.armor", rabbitArmor);
+        rabbitToastArmorToughness = getDoubleIfSet("mobs.rabbit.attribute.toast.armor_toughness", rabbitArmorToughness);
+    }
 }
\ No newline at end of file
