From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: AlphaKR93 <alphakr93@outlook.com>
Date: Sun, 2 Oct 2022 21:26:45 +0900
Subject: [PATCH] Force despawn API


diff --git a/src/main/java/net/minecraft/world/entity/Entity.java b/src/main/java/net/minecraft/world/entity/Entity.java
index 1d7d547507567ea2b4f7deb2bdb06229b2886ba4..478832dbbb9ae72788ce82e9b6d7675b9685f80d 100644
--- a/src/main/java/net/minecraft/world/entity/Entity.java
+++ b/src/main/java/net/minecraft/world/entity/Entity.java
@@ -4432,6 +4432,8 @@ public abstract class Entity implements Nameable, EntityAccess, CommandSource {
 
     public void checkDespawn() {}
 
+    public void setForceDespawn(boolean force) {} // Prismarine
+
     public Vec3 getRopeHoldPosition(float delta) {
         return this.getPosition(delta).add(0.0D, (double) this.eyeHeight * 0.7D, 0.0D);
     }
diff --git a/src/main/java/net/minecraft/world/entity/Mob.java b/src/main/java/net/minecraft/world/entity/Mob.java
index 920318f80256632a8b6ac430e18ea470dfbfba37..fbf4db7ccf25af0765132c3638699b8fee50aece 100644
--- a/src/main/java/net/minecraft/world/entity/Mob.java
+++ b/src/main/java/net/minecraft/world/entity/Mob.java
@@ -136,6 +136,7 @@ public abstract class Mob extends LivingEntity {
 
     public int ticksSinceLastInteraction; // Purpur
     public boolean aware = true; // CraftBukkit
+    public boolean forceDespawn = false; // Prismarine
 
     protected Mob(EntityType<? extends Mob> type, Level world) {
         super(type, world);
@@ -838,7 +839,7 @@ public abstract class Mob extends LivingEntity {
     public void checkDespawn() {
         if (this.level.getDifficulty() == Difficulty.PEACEFUL && this.shouldDespawnInPeaceful()) {
             this.discard();
-        } else if (!this.isPersistenceRequired() && !this.requiresCustomPersistence()) {
+        } else if (!this.isPersistenceRequired() && !this.requiresCustomPersistence() || this.forceDespawn) { // Prismarine
             // Paper start - optimise checkDespawn
             Player entityhuman = this.level.findNearbyPlayer(this, level.paperConfig().entities.spawning.despawnRanges.get(this.getType().getCategory()).hard() + 1, EntitySelector.affectsSpawning); // Paper
             // Paper end - optimise checkDespawn
@@ -870,6 +871,13 @@ public abstract class Mob extends LivingEntity {
         }
     }
 
+    // Prismarine start
+    @Override
+    public void setForceDespawn(boolean force) {
+        this.forceDespawn = force;
+    }
+    // Prismarine end
+
     @Override
     protected final void serverAiStep() {
         ++this.noActionTime;
diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
index 3d62afb3add239c7ba61551f6729611ffc80dc9c..a76fcebcf75b0bd8d330175db0ae1fd21d6c727f 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
@@ -687,6 +687,13 @@ public abstract class CraftEntity implements org.bukkit.entity.Entity {
         this.entity.discard();
     }
 
+    // Prismairne start
+    @Override
+    public void setRemove(boolean force) {
+        this.entity.setForceDespawn(force);
+    }
+    // Prismarine end
+
     @Override
     public boolean isDead() {
         return !this.entity.isAlive();
